--- gtk/src/gtk_signals.defs.orig	2013-09-11 22:05:42.655978022 +0200
+++ gtk/src/gtk_signals.defs	2013-09-11 22:07:53.554448890 +0200
@@ -2455,21 +2455,21 @@
   (construct-only #f)
 )
 
 ;; From GtkClipboard
 
 (define-signal owner-change
   (of-object "GtkClipboard")
   (return-type "void")
   (when "first")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventOwnerChange*" "p0")
   )
 )
 
 ;; From GtkColorButton
 
 (define-signal color-set
   (of-object "GtkColorButton")
   (return-type "void")
   (when "first")
 )
@@ -2519,21 +2519,21 @@
   (construct-only #f)
 )
 
 ;; From GtkColorChooser
 
 (define-signal color-activated
   (of-object "GtkColorChooser")
   (return-type "void")
   (when "first")
   (parameters
-    '("GdkRGBA*" "p0")
+    '("const-GdkRGBA*" "p0")
   )
 )
 
 (define-property rgba
   (of-object "GtkColorChooser")
   (prop-type "GParamBoxed")
   (docs "Current color, as a GdkRGBA")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -2674,21 +2674,21 @@
   (of-object "GtkComboBox")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkScrollType" "p0")
   )
 )
 
 (define-signal format-entry-text
   (of-object "GtkComboBox")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (when "last")
   (parameters
     '("const-gchar*" "p0")
   )
 )
 
 (define-property model
   (of-object "GtkComboBox")
   (prop-type "GParamObject")
   (docs "The model for the combo box")
@@ -3119,31 +3119,31 @@
     '("const-gchar*" "p0")
   )
 )
 
 (define-signal icon-press
   (of-object "GtkEntry")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkEntryIconPosition" "p0")
-    '("GdkEvent*" "p1")
+    '("const-GdkEventButton*" "p1")
   )
 )
 
 (define-signal icon-release
   (of-object "GtkEntry")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkEntryIconPosition" "p0")
-    '("GdkEvent*" "p1")
+    '("const-GdkEventButton*" "p1")
   )
 )
 
 (define-property buffer
   (of-object "GtkEntry")
   (prop-type "GParamObject")
   (docs "Text buffer object which actually stores entry text")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -5219,21 +5219,21 @@
   (of-object "GtkMenuItem")
   (return-type "void")
   (when "first")
 )
 
 (define-signal toggle-size-request
   (of-object "GtkMenuItem")
   (return-type "void")
   (when "first")
   (parameters
-    '("gpointer" "p0")
+    '("int*" "p0")
   )
 )
 
 (define-signal toggle-size-allocate
   (of-object "GtkMenuItem")
   (return-type "void")
   (when "first")
   (parameters
     '("gint" "p0")
   )
@@ -5805,32 +5805,32 @@
   (construct-only #f)
 )
 
 ;; From GtkPlacesSidebar
 
 (define-signal populate-popup
   (of-object "GtkPlacesSidebar")
   (return-type "void")
   (when "first")
   (parameters
-    '("GObject*" "p0")
-    '("GObject*" "p1")
-    '("GObject*" "p2")
+    '("GtkMenu*" "p0")
+    '("GFile*" "p1")
+    '("GVolume*" "p2")
   )
 )
 
 (define-signal open-location
   (of-object "GtkPlacesSidebar")
   (return-type "void")
   (when "first")
   (parameters
-    '("GObject*" "p0")
+    '("GFile*" "p0")
     '("GtkPlacesOpenFlags" "p1")
   )
 )
 
 (define-signal show-error-message
   (of-object "GtkPlacesSidebar")
   (return-type "void")
   (when "first")
   (parameters
     '("const-gchar*" "p0")
@@ -5843,41 +5843,41 @@
   (return-type "void")
   (when "first")
 )
 
 (define-signal drag-action-requested
   (of-object "GtkPlacesSidebar")
   (return-type "gint")
   (when "last")
   (parameters
     '("GdkDragContext*" "p0")
-    '("GObject*" "p1")
-    '("gpointer" "p2")
+    '("GFile*" "p1")
+    '("GList*" "p2")
   )
 )
 
 (define-signal drag-action-ask
   (of-object "GtkPlacesSidebar")
   (return-type "gint")
   (when "last")
   (parameters
     '("gint" "p0")
   )
 )
 
 (define-signal drag-perform-drop
   (of-object "GtkPlacesSidebar")
   (return-type "void")
   (when "first")
   (parameters
-    '("GObject*" "p0")
-    '("gpointer" "p1")
+    '("GFile*" "p0")
+    '("GList*" "p1")
     '("gint" "p2")
   )
 )
 
 (define-property location
   (of-object "GtkPlacesSidebar")
   (prop-type "GParamObject")
   (docs "The location to highlight in the sidebar")
   (readable #t)
   (writable #t)
@@ -6160,21 +6160,21 @@
 )
 
 (define-signal status-changed
   (of-object "GtkPrintOperation")
   (return-type "void")
   (when "last")
 )
 
 (define-signal create-custom-widget
   (of-object "GtkPrintOperation")
-  (return-type "GObject*")
+  (return-type "GtkWidget*")
   (when "last")
 )
 
 (define-signal update-custom-widget
   (of-object "GtkPrintOperation")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkWidget*" "p0")
     '("GtkPageSetup*" "p1")
@@ -6922,21 +6922,21 @@
   (docs "Whether the child is revealed and the animation target reached")
   (readable #t)
   (writable #f)
   (construct-only #f)
 )
 
 ;; From GtkScale
 
 (define-signal format-value
   (of-object "GtkScale")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (when "last")
   (parameters
     '("gdouble" "p0")
   )
 )
 
 (define-property digits
   (of-object "GtkScale")
   (prop-type "GParamInt")
   (docs "The number of decimal places that are displayed in the value")
@@ -7318,21 +7318,21 @@
   (prop-type "GParamString")
   (docs "Name of default font to use")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
 (define-property gtk-icon-sizes
   (of-object "GtkSettings")
   (prop-type "GParamString")
-  (docs "List of icon sizes (gtk-menu=16,16:gtk-button=20,20...")
+  (docs "List of icon sizes: gtk-menu=16,16:gtk-button=20,20...")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
 (define-property gtk-modules
   (of-object "GtkSettings")
   (prop-type "GParamString")
   (docs "List of currently active GTK modules")
   (readable #t)
@@ -7931,21 +7931,21 @@
   (of-object "GtkSpinButton")
   (return-type "void")
   (when "last")
 )
 
 (define-signal input
   (of-object "GtkSpinButton")
   (return-type "gint")
   (when "last")
   (parameters
-    '("gpointer" "p0")
+    '("double*" "p0")
   )
 )
 
 (define-signal output
   (of-object "GtkSpinButton")
   (return-type "gboolean")
   (when "last")
 )
 
 (define-signal wrapped
@@ -8057,39 +8057,39 @@
   )
 )
 
 ;; From GtkStatusIcon
 
 (define-signal button-press-event
   (of-object "GtkStatusIcon")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal button-release-event
   (of-object "GtkStatusIcon")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal scroll-event
   (of-object "GtkStatusIcon")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventScroll*" "p0")
   )
 )
 
 (define-signal query-tooltip
   (of-object "GtkStatusIcon")
   (return-type "gboolean")
   (when "last")
   (parameters
     '("gint" "p0")
     '("gint" "p1")
@@ -8097,21 +8097,21 @@
     '("GtkTooltip*" "p3")
   )
 )
 
 (define-signal popup-menu
   (of-object "GtkStatusIcon")
   (return-type "void")
   (when "first")
   (parameters
     '("guint" "p0")
-    '("guint" "p1")
+    '("guint32" "p1")
   )
 )
 
 (define-signal activate
   (of-object "GtkStatusIcon")
   (return-type "void")
   (when "first")
 )
 
 (define-signal size-changed
@@ -8713,53 +8713,53 @@
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
 )
 
 (define-signal mark-set
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
   (parameters
-    '("GtkTextIter*" "p0")
+    '("const-GtkTextIter*" "p0")
     '("GtkTextMark*" "p1")
   )
 )
 
 (define-signal mark-deleted
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkTextMark*" "p0")
   )
 )
 
 (define-signal apply-tag
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal remove-tag
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal begin-user-action
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
 )
 
 (define-signal end-user-action
@@ -8853,21 +8853,21 @@
 
 ;; From GtkTextTag
 
 (define-signal event
   (of-object "GtkTextTag")
   (return-type "gboolean")
   (when "last")
   (parameters
     '("GObject*" "p0")
     '("GdkEvent*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-property name
   (of-object "GtkTextTag")
   (prop-type "GParamString")
   (docs "Name used to refer to the text tag. NULL for anonymous tags")
   (readable #t)
   (writable #t)
   (construct-only #t)
@@ -9846,21 +9846,21 @@
   )
 )
 
 (define-signal rows-reordered
   (of-object "GtkTreeModel")
   (return-type "void")
   (when "first")
   (parameters
     '("GtkTreePath*" "p0")
     '("GtkTreeIter*" "p1")
-    '("gpointer" "p2")
+    '("gint*" "p2")
   )
 )
 
 ;; From GtkTreeSelection
 
 (define-signal changed
   (of-object "GtkTreeSelection")
   (return-type "void")
   (when "first")
 )
@@ -10595,30 +10595,30 @@
   (parameters
     '("gboolean" "p0")
   )
 )
 
 (define-signal child-notify
   (of-object "GtkWidget")
   (return-type "void")
   (when "first")
   (parameters
-    '("GParam" "p0")
+    '("GParamSpec*" "p0")
   )
 )
 
 (define-signal draw
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("CairoContext*" "p0")
+    '("cairo_t*" "p0")
   )
 )
 
 (define-signal mnemonic-activate
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
     '("gboolean" "p0")
   )
@@ -10673,198 +10673,198 @@
   (parameters
     '("GdkEvent*" "p0")
   )
 )
 
 (define-signal button-press-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal button-release-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal touch-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventTouch*" "p0")
   )
 )
 
 (define-signal scroll-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventScroll*" "p0")
   )
 )
 
 (define-signal motion-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventMotion*" "p0")
   )
 )
 
 (define-signal composited-changed
   (of-object "GtkWidget")
   (return-type "void")
   (when "last")
 )
 
 (define-signal delete-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal destroy-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal key-press-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventKey*" "p0")
   )
 )
 
 (define-signal key-release-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventKey*" "p0")
   )
 )
 
 (define-signal enter-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventCrossing*" "p0")
   )
 )
 
 (define-signal leave-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventCrossing*" "p0")
   )
 )
 
 (define-signal configure-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventConfigure*" "p0")
   )
 )
 
 (define-signal focus-in-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventFocus*" "p0")
   )
 )
 
 (define-signal focus-out-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventFocus*" "p0")
   )
 )
 
 (define-signal map-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal unmap-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal property-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventProperty*" "p0")
   )
 )
 
 (define-signal selection-clear-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventSelection*" "p0")
   )
 )
 
 (define-signal selection-request-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventSelection*" "p0")
   )
 )
 
 (define-signal selection-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventSelection*" "p0")
   )
 )
 
 (define-signal selection-received
   (of-object "GtkWidget")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkSelectionData*" "p0")
     '("guint" "p1")
@@ -10880,30 +10880,30 @@
     '("guint" "p1")
     '("guint" "p2")
   )
 )
 
 (define-signal proximity-in-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventProximity*" "p0")
   )
 )
 
 (define-signal proximity-out-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventProximity*" "p0")
   )
 )
 
 (define-signal drag-leave
   (of-object "GtkWidget")
   (return-type "void")
   (when "last")
   (parameters
     '("GdkDragContext*" "p0")
     '("guint" "p1")
@@ -10995,48 +10995,48 @@
     '("guint" "p4")
     '("guint" "p5")
   )
 )
 
 (define-signal visibility-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventVisibility*" "p0")
   )
 )
 
 (define-signal window-state-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventWindowState*" "p0")
   )
 )
 
 (define-signal damage-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventExpose*" "p0")
   )
 )
 
 (define-signal grab-broken-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventGrabBroken*" "p0")
   )
 )
 
 (define-signal query-tooltip
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
     '("gint" "p0")
     '("gint" "p1")
