--- a/gtk/src/gtk_signals.defs
+++ b/gtk/src/gtk_signals.defs
@@ -533,7 +533,7 @@
   (when "last")
   (parameters
     '("GtkEntryIconPosition" "icon_pos")
+    '("const-GdkEventButton*" "event")
-    '("GdkEventButton*" "event")
   )
 )
 
@@ -543,7 +543,7 @@
   (when "last")
   (parameters
     '("GtkEntryIconPosition" "icon_pos")
+    '("const-GdkEventButton*" "event")
-    '("GdkEventButton*" "event")
   )
 )
 
@@ -1328,7 +1328,7 @@
   (return-type "void")
   (when "first")
   (parameters
+    '("int*" "object")
-    '("gpointer" "object")
   )
 )
 
@@ -1699,7 +1699,7 @@
 
 (define-signal create-custom-widget
   (of-object "GtkPrintOperation")
+  (return-type "GtkWidget*")
-  (return-type "GObject*")
   (when "last")
 )
 
@@ -2141,8 +2141,8 @@
   (when "last")
   (parameters
     '("GtkTextTag*" "tag")
+    '("const-GtkTextIter*" "start")
+    '("const-GtkTextIter*" "end")
-    '("GtkTextIter*" "start")
-    '("GtkTextIter*" "end")
   )
 )
 
@@ -2219,7 +2219,7 @@
   (return-type "void")
   (when "last")
   (parameters
+    '("const-GtkTextIter*" "location")
-    '("GtkTextIter*" "location")
     '("GtkTextMark*" "mark")
   )
 )
@@ -2245,8 +2245,8 @@
   (when "last")
   (parameters
     '("GtkTextTag*" "tag")
+    '("const-GtkTextIter*" "start")
+    '("const-GtkTextIter*" "end")
-    '("GtkTextIter*" "start")
-    '("GtkTextIter*" "end")
   )
 )
 
@@ -2257,7 +2257,7 @@
   (parameters
     '("GObject*" "object")
     '("GdkEvent*" "event")
+    '("const-GtkTextIter*" "iter")
-    '("GtkTextIter*" "iter")
   )
 )
 
@@ -2742,7 +2742,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
+    '("GdkEventAny*" "event")
-    '("GdkEvent*" "event")
   )
 )
 
@@ -3159,7 +3159,7 @@
   (return-type "void")
   (when "last")
   (parameters
+    '("GtkSelectionData*" "selection_data")
-    '("GtkSelectionData*" "data1")
     '("guint" "time")
   )
 )
@@ -3193,7 +3193,7 @@
   (return-type "void")
   (when "first")
   (parameters
+    '("GdkRectangle*" "allocation")
-    '("GtkAllocation" "allocation")
   )
 )
 
@@ -3235,7 +3235,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
+    '("GdkEventTouch*" "object")
-    '("GdkEvent*" "object")
   )
 )
 
@@ -3331,7 +3331,7 @@
   (return-type "void")
   (when "first")
   (parameters
+    '("const-GdkRGBA*" "color")
-    '("GdkRGBA*" "color")
   )
 )
 
@@ -3478,7 +3478,7 @@
   (parameters
     '("GtkTreePath*" "path")
     '("GtkTreeIter*" "iter")
+    '("int*" "new_order")
-    '("gpointer" "new_order")
   )
 )
 
