--- ./../../gtk/src/gtk_signals.defs.orig	2017-12-16 10:19:02.792433346 +0100
+++ ./../../gtk/src/gtk_signals.defs	2017-12-16 10:25:33.054013331 +0100
@@ -2276,11 +2276,11 @@
 (define-signal color-activated
   (of-object "GtkColorChooser")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GdkRGBA*" "p0")
+    '("const-GdkRGBA*" "p0")
   )
 )
 
 (define-property rgba
   (of-object "GtkColorChooser")
@@ -2342,11 +2342,11 @@
   (flags "Run Last, Action")
 )
 
 (define-signal format-entry-text
   (of-object "GtkComboBox")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (flags "Run Last")
   (parameters
     '("const-gchar*" "p0")
   )
 )
@@ -2570,11 +2570,11 @@
   (of-object "GtkCssProvider")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkCssSection*" "p0")
-    '("GError*" "p1")
+    '("const-GError*" "p1")
   )
 )
 
 ;; From GtkDialog
 
@@ -2735,21 +2735,21 @@
   (of-object "GtkEntry")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkEntryIconPosition" "p0")
-    '("GdkEvent*" "p1")
+    '("const-GdkEventButton*" "p1")
   )
 )
 
 (define-signal icon-release
   (of-object "GtkEntry")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkEntryIconPosition" "p0")
-    '("GdkEvent*" "p1")
+    '("const-GdkEventButton*" "p1")
   )
 )
 
 (define-signal preedit-changed
   (of-object "GtkEntry")
@@ -5717,11 +5717,11 @@
 (define-signal toggle-size-request
   (of-object "GtkMenuItem")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("gpointer" "p0")
+    '("int*" "p0")
   )
 )
 
 (define-signal toggle-size-allocate
   (of-object "GtkMenuItem")
@@ -6384,11 +6384,11 @@
 (define-signal open-location
   (of-object "GtkPlacesSidebar")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GObject*" "p0")
+    '("GFile*" "p0")
     '("GtkPlacesOpenFlags" "p1")
   )
 )
 
 (define-signal show-error-message
@@ -6411,12 +6411,12 @@
   (of-object "GtkPlacesSidebar")
   (return-type "gint")
   (flags "Run Last")
   (parameters
     '("GdkDragContext*" "p0")
-    '("GObject*" "p1")
-    '("gpointer" "p2")
+    '("GFile*" "p1")
+    '("GList*" "p2")
   )
 )
 
 (define-signal drag-action-ask
   (of-object "GtkPlacesSidebar")
@@ -6430,12 +6430,12 @@
 (define-signal drag-perform-drop
   (of-object "GtkPlacesSidebar")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GObject*" "p0")
-    '("gpointer" "p1")
+    '("GFile*" "p0")
+    '("GList*" "p1")
     '("gint" "p2")
   )
 )
 
 (define-signal show-other-locations-with-flags
@@ -6881,11 +6881,11 @@
   )
 )
 
 (define-signal create-custom-widget
   (of-object "GtkPrintOperation")
-  (return-type "GObject*")
+  (return-type "GtkWidget*")
   (flags "Run Last")
 )
 
 (define-signal update-custom-widget
   (of-object "GtkPrintOperation")
@@ -7623,11 +7623,11 @@
 
 ;; From GtkScale
 
 (define-signal format-value
   (of-object "GtkScale")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (flags "Run Last")
   (parameters
     '("gdouble" "p0")
   )
 )
@@ -8815,11 +8815,11 @@
 (define-signal input
   (of-object "GtkSpinButton")
   (return-type "gint")
   (flags "Run Last")
   (parameters
-    '("gpointer" "p0")
+    '("double*" "p0")
   )
 )
 
 (define-signal output
   (of-object "GtkSpinButton")
@@ -9466,11 +9466,11 @@
 (define-signal mark-set
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
-    '("GtkTextIter*" "p0")
+    '("const-GtkTextIter*" "p0")
     '("GtkTextMark*" "p1")
   )
 )
 
 (define-signal mark-deleted
@@ -9486,23 +9486,23 @@
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal remove-tag
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal begin-user-action
   (of-object "GtkTextBuffer")
@@ -10760,11 +10760,11 @@
   (return-type "void")
   (flags "Run First")
   (parameters
     '("GtkTreePath*" "p0")
     '("GtkTreeIter*" "p1")
-    '("gpointer" "p2")
+    '("gint*" "p2")
   )
 )
 
 ;; From GtkTreeSelection
 
@@ -11384,11 +11384,11 @@
 (define-signal size-allocate
   (of-object "GtkWidget")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GdkRectangle*" "p0")
+    '("const-GdkRectangle*" "p0")
     '("gint" "p1")
     '("GdkRectangle*" "p2")
   )
 )
 
@@ -11438,20 +11438,20 @@
   (of-object "GtkWidget")
   (return-type "void")
   (flags "Run First, No Recurse, No Hooks")
   (detailed #t)
   (parameters
-    '("GParam" "p0")
+    '("GParamSpec*" "p0")
   )
 )
 
 (define-signal draw
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("CairoContext*" "p0")
+    '("cairo_t*" "p0")
   )
 )
 
 (define-signal mnemonic-activate
   (of-object "GtkWidget")
@@ -11519,172 +11519,172 @@
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal button-release-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal touch-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventTouch*" "p0")
   )
 )
 
 (define-signal scroll-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventScroll*" "p0")
   )
 )
 
 (define-signal motion-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventMotion*" "p0")
   )
 )
 
 (define-signal delete-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal destroy-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal key-press-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventKey*" "p0")
   )
 )
 
 (define-signal key-release-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventKey*" "p0")
   )
 )
 
 (define-signal enter-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventCrossing*" "p0")
   )
 )
 
 (define-signal leave-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventCrossing*" "p0")
   )
 )
 
 (define-signal configure-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventConfigure*" "p0")
   )
 )
 
 (define-signal focus-in-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventFocus*" "p0")
   )
 )
 
 (define-signal focus-out-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventFocus*" "p0")
   )
 )
 
 (define-signal map-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal unmap-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal proximity-in-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventProximity*" "p0")
   )
 )
 
 (define-signal proximity-out-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (deprecated #t)
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventProximity*" "p0")
   )
 )
 
 (define-signal drag-leave
   (of-object "GtkWidget")
@@ -11782,11 +11782,11 @@
 (define-signal grab-broken-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventGrabBroken*" "p0")
   )
 )
 
 (define-signal query-tooltip
   (of-object "GtkWidget")
