? temp.patch
Index: gtk_signals.defs
===================================================================
RCS file: /cvs/gnome/gtkmm/gtk/src/gtk_signals.defs,v
retrieving revision 1.18
diff -u -p -r1.18 gtk_signals.defs
--- gtk_signals.defs	17 Aug 2005 21:22:38 -0000	1.18
+++ gtk_signals.defs	17 Aug 2005 21:24:03 -0000
@@ -156,7 +156,7 @@
 (define-property no-show-all
   (of-object "GtkAboutDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -342,6 +342,15 @@
   (construct-only #f)
 )
 
+(define-property urgency-hint
+  (of-object "GtkAboutDialog")
+  (prop-type "GParamBoolean")
+  (docs "TRUE if the window should be brought to the user's attention.")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property accept-focus
   (of-object "GtkAboutDialog")
   (prop-type "GParamBoolean")
@@ -408,7 +417,7 @@
 (define-property name
   (of-object "GtkAboutDialog")
   (prop-type "GParamString")
-  (docs "The name of the program. If this is not set, it defaults to g_get_application_name")
+  (docs "The name of the program. If this is not set, it defaults to g_get_application_name()")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -507,7 +516,7 @@
 (define-property logo
   (of-object "GtkAboutDialog")
   (prop-type "GParamObject")
-  (docs "A logo for the about box. If this is not set, it defaults to gtk_window_get_default_icon_list")
+  (docs "A logo for the about box. If this is not set, it defaults to gtk_window_get_default_icon_list()")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -722,7 +731,7 @@
 (define-property no-show-all
   (of-object "GtkAccelLabel")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -785,7 +794,7 @@
 (define-property use-markup
   (of-object "GtkAccelLabel")
   (prop-type "GParamBoolean")
-  (docs "The text of the label includes XML markup. See pango_parse_markup")
+  (docs "The text of the label includes XML markup. See pango_parse_markup()")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -1371,7 +1380,7 @@
 (define-property no-show-all
   (of-object "GtkAlignment")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -1643,7 +1652,7 @@
 (define-property no-show-all
   (of-object "GtkArrow")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -1870,7 +1879,7 @@
 (define-property no-show-all
   (of-object "GtkAspectFrame")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -2160,7 +2169,7 @@
 (define-property no-show-all
   (of-object "GtkBin")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -2360,7 +2369,7 @@
 (define-property no-show-all
   (of-object "GtkBox")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -2614,7 +2623,7 @@
 (define-property no-show-all
   (of-object "GtkButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -2886,7 +2895,7 @@
 (define-property no-show-all
   (of-object "GtkButtonBox")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -3155,7 +3164,7 @@
 (define-property no-show-all
   (of-object "GtkCalendar")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -3391,7 +3400,7 @@
 (define-property no-show-all
   (of-object "GtkCheckButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -3508,7 +3517,7 @@
 (define-property inconsistent
   (of-object "GtkCheckButton")
   (prop-type "GParamBoolean")
-  (docs "If the toggle button is in an in-between state")
+  (docs "If the toggle button is in an "in between" state")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -3696,7 +3705,7 @@
 (define-property no-show-all
   (of-object "GtkCheckMenuItem")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -3741,7 +3750,7 @@
 (define-property inconsistent
   (of-object "GtkCheckMenuItem")
   (prop-type "GParamBoolean")
-  (docs "Whether to display an inconsistent state")
+  (docs "Whether to display an "inconsistent" state")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -4063,7 +4072,7 @@
 (define-property no-show-all
   (of-object "GtkCList")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -6264,7 +6273,7 @@
 (define-property no-show-all
   (of-object "GtkColorButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -6578,7 +6587,7 @@
 (define-property no-show-all
   (of-object "GtkColorSelection")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -6832,7 +6841,7 @@
 (define-property no-show-all
   (of-object "GtkCombo")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -7101,7 +7110,7 @@
 (define-property no-show-all
   (of-object "GtkComboBox")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -7406,7 +7415,7 @@
 (define-property no-show-all
   (of-object "GtkContainer")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -7664,7 +7673,7 @@
 (define-property no-show-all
   (of-object "GtkCTree")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -7990,7 +7999,7 @@
 (define-property no-show-all
   (of-object "GtkCellView")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -8196,7 +8205,7 @@
 (define-property no-show-all
   (of-object "GtkCurve")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -8429,7 +8438,7 @@
 (define-property no-show-all
   (of-object "GtkDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -8854,7 +8863,7 @@
 (define-property no-show-all
   (of-object "GtkDrawingArea")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -9102,7 +9111,7 @@
 (define-property no-show-all
   (of-object "GtkEntry")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -9147,7 +9156,7 @@
 (define-property visibility
   (of-object "GtkEntry")
   (prop-type "GParamBoolean")
-  (docs "FALSE displays the invisible char instead of the actual text (password mode)")
+  (docs "FALSE displays the "invisible char" instead of the actual text (password mode)")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -9165,7 +9174,7 @@
 (define-property invisible-char
   (of-object "GtkEntry")
   (prop-type "GParamUnichar")
-  (docs "The character to use when masking entry contents (in password mode)")
+  (docs "The character to use when masking entry contents (in "password mode")")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -9476,7 +9485,7 @@
 (define-property no-show-all
   (of-object "GtkEventBox")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -9700,7 +9709,7 @@
 (define-property no-show-all
   (of-object "GtkExpander")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -9763,7 +9772,7 @@
 (define-property use-markup
   (of-object "GtkExpander")
   (prop-type "GParamBoolean")
-  (docs "The text of the label includes XML markup. See pango_parse_markup")
+  (docs "The text of the label includes XML markup. See pango_parse_markup()")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -10064,7 +10073,7 @@
 (define-property no-show-all
   (of-object "GtkFileChooserButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -10408,7 +10417,7 @@
 (define-property no-show-all
   (of-object "GtkFileChooserDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -10932,7 +10941,7 @@
 (define-property no-show-all
   (of-object "GtkFileChooserWidget")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -11150,7 +11159,7 @@
 (define-property no-show-all
   (of-object "GtkFileSelection")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -11602,7 +11611,7 @@
 (define-property no-show-all
   (of-object "GtkFixed")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -11808,7 +11817,7 @@
 (define-property no-show-all
   (of-object "GtkFontButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -12134,7 +12143,7 @@
 (define-property no-show-all
   (of-object "GtkFontSelection")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -12379,7 +12388,7 @@
 (define-property no-show-all
   (of-object "GtkFontSelectionDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -12566,15 +12575,6 @@
 )
 
 (define-property urgency-hint
-  (of-object "GtkFileChooserDialog")
-  (prop-type "GParamBoolean")
-  (docs "TRUE if the window should be brought to the user's attention.")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property urgency-hint
   (of-object "GtkFontSelectionDialog")
   (prop-type "GParamBoolean")
   (docs "TRUE if the window should be brought to the user's attention.")
@@ -12813,7 +12813,7 @@
 (define-property no-show-all
   (of-object "GtkFrame")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -13085,7 +13085,7 @@
 (define-property no-show-all
   (of-object "GtkHandleBox")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -13338,7 +13338,7 @@
 (define-property no-show-all
   (of-object "GtkImage")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -13420,7 +13420,7 @@
   (of-object "GtkImage")
   (prop-type "GParamString")
   (docs "Filename to load and display")
-  (readable #f)
+  (readable #t)
   (writable #t)
   (construct-only #f)
 )
@@ -13673,7 +13673,7 @@
 (define-property no-show-all
   (of-object "GtkInputDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -14098,7 +14098,7 @@
 (define-property no-show-all
   (of-object "GtkInvisible")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -14298,7 +14298,7 @@
 (define-property no-show-all
   (of-object "GtkItem")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -14563,7 +14563,7 @@
 (define-property no-show-all
   (of-object "GtkIconView")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -14906,7 +14906,7 @@
 (define-property no-show-all
   (of-object "GtkLabel")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -14969,7 +14969,7 @@
 (define-property use-markup
   (of-object "GtkLabel")
   (prop-type "GParamBoolean")
-  (docs "The text of the label includes XML markup. See pango_parse_markup")
+  (docs "The text of the label includes XML markup. See pango_parse_markup()")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -15278,7 +15278,7 @@
 (define-property no-show-all
   (of-object "GtkLayout")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -15538,7 +15538,7 @@
 (define-property no-show-all
   (of-object "GtkList")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -15820,7 +15820,7 @@
 (define-property no-show-all
   (of-object "GtkListItem")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -16029,7 +16029,7 @@
 (define-property no-show-all
   (of-object "GtkMenu")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -16256,7 +16256,7 @@
 (define-property no-show-all
   (of-object "GtkMenuBar")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -16335,7 +16335,7 @@
   (return-type "void")
   (when "first")
   (parameters
-    '("int*" "p0")
+    '("gpointer" "p0")
   )
 )
 
@@ -16513,7 +16513,7 @@
 (define-property no-show-all
   (of-object "GtkMenuItem")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -16758,7 +16758,7 @@
 (define-property no-show-all
   (of-object "GtkMenuShell")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -16973,7 +16973,7 @@
 (define-property no-show-all
   (of-object "GtkMenuToolButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -17263,7 +17263,7 @@
 (define-property no-show-all
   (of-object "GtkMessageDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -17449,19 +17449,19 @@
   (construct-only #f)
 )
 
-(define-property accept-focus
+(define-property urgency-hint
   (of-object "GtkMessageDialog")
   (prop-type "GParamBoolean")
-  (docs "TRUE if the window should receive the input focus.")
+  (docs "TRUE if the window should be brought to the user's attention.")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property urgency-hint
+(define-property accept-focus
   (of-object "GtkMessageDialog")
   (prop-type "GParamBoolean")
-  (docs "TRUE if the window should be brought to the user's attention.")
+  (docs "TRUE if the window should receive the input focus.")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -17706,7 +17706,7 @@
 (define-property no-show-all
   (of-object "GtkMisc")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -17764,7 +17764,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("GtkNotebookPage*" "p0")
+    '("gpointer" "p0")
     '("guint" "p1")
   )
 )
@@ -17961,7 +17961,7 @@
 (define-property no-show-all
   (of-object "GtkNotebook")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -18274,7 +18274,7 @@
 (define-property no-show-all
   (of-object "GtkOptionMenu")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -18600,7 +18600,7 @@
 (define-property no-show-all
   (of-object "GtkPaned")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -18654,7 +18654,7 @@
 (define-property min-position
   (of-object "GtkPaned")
   (prop-type "GParamInt")
-  (docs "Smallest possible value for the position property")
+  (docs "Smallest possible value for the "position" property")
   (readable #t)
   (writable #f)
   (construct-only #f)
@@ -18663,7 +18663,7 @@
 (define-property max-position
   (of-object "GtkPaned")
   (prop-type "GParamInt")
-  (docs "Largest possible value for the position property")
+  (docs "Largest possible value for the "position" property")
   (readable #t)
   (writable #f)
   (construct-only #f)
@@ -18836,7 +18836,7 @@
 (define-property no-show-all
   (of-object "GtkPixmap")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -19051,7 +19051,7 @@
 (define-property no-show-all
   (of-object "GtkPlug")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -19467,7 +19467,7 @@
 (define-property no-show-all
   (of-object "GtkPreview")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -19649,7 +19649,7 @@
 (define-property no-show-all
   (of-object "GtkProgress")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -19858,7 +19858,7 @@
 (define-property no-show-all
   (of-object "GtkProgressBar")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -20163,7 +20163,7 @@
 (define-property no-show-all
   (of-object "GtkRadioButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -20280,7 +20280,7 @@
 (define-property inconsistent
   (of-object "GtkRadioButton")
   (prop-type "GParamBoolean")
-  (docs "If the toggle button is in an in-between state")
+  (docs "If the toggle button is in an "in between" state")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -20471,7 +20471,7 @@
 (define-property no-show-all
   (of-object "GtkRadioToolButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -20743,7 +20743,7 @@
 (define-property value
   (of-object "GtkRadioAction")
   (prop-type "GParamInt")
-  (docs "The value returned by gtk_radio_action_get_current_value when this action is the current action of its group.")
+  (docs "The value returned by gtk_radio_action_get_current_value() when this action is the current action of its group.")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -20931,7 +20931,7 @@
 (define-property no-show-all
   (of-object "GtkRadioMenuItem")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -20976,7 +20976,7 @@
 (define-property inconsistent
   (of-object "GtkRadioMenuItem")
   (prop-type "GParamBoolean")
-  (docs "Whether to display an inconsistent state")
+  (docs "Whether to display an "inconsistent" state")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -21201,7 +21201,7 @@
 (define-property no-show-all
   (of-object "GtkRange")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -21401,7 +21401,7 @@
 (define-property no-show-all
   (of-object "GtkRuler")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -21456,7 +21456,7 @@
 
 (define-signal format-value
   (of-object "GtkScale")
-  (return-type "gchar*")
+  (return-type "const-gchar*")
   (when "last")
   (parameters
     '("gdouble" "p0")
@@ -21628,7 +21628,7 @@
 (define-property no-show-all
   (of-object "GtkScale")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -21855,7 +21855,7 @@
 (define-property no-show-all
   (of-object "GtkScrollbar")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -22074,7 +22074,7 @@
 (define-property no-show-all
   (of-object "GtkScrolledWindow")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -22328,7 +22328,7 @@
 (define-property no-show-all
   (of-object "GtkSeparator")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -22501,7 +22501,7 @@
 (define-property no-show-all
   (of-object "GtkSeparatorToolItem")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -22674,7 +22674,7 @@
 (define-property gtk-icon-sizes
   (of-object "GtkSettings")
   (prop-type "GParamString")
-  (docs "List of icon sizes gtk-menu=16,16;gtk-button=20,20...")
+  (docs "List of icon sizes (gtk-menu=16,16:gtk-button=20,20...")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -22710,7 +22710,7 @@
 (define-property gtk-xft-hintstyle
   (of-object "GtkSettings")
   (prop-type "GParamString")
-  (docs "What degree of hinting to use; none, slight, medium, or full")
+  (docs "What degree of hinting to use; hintnone, hintslight, hintmedium, or hintfull")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -22802,7 +22802,7 @@
 (define-property mode
   (of-object "GtkSizeGroup")
   (prop-type "GParamEnum")
-  (docs "The directions in which the size group effects the requested sizes of its component widgets")
+  (docs "The directions in which the size group affects the requested sizes of its component widgets")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -22996,7 +22996,7 @@
 (define-property no-show-all
   (of-object "GtkSocket")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -23036,7 +23036,7 @@
   (return-type "gint")
   (when "last")
   (parameters
-    '("gdouble*" "p0")
+    '("gpointer" "p0")
   )
 )
 
@@ -23226,7 +23226,7 @@
 (define-property no-show-all
   (of-object "GtkSpinButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -23271,7 +23271,7 @@
 (define-property visibility
   (of-object "GtkSpinButton")
   (prop-type "GParamBoolean")
-  (docs "FALSE displays the invisible char instead of the actual text (password mode)")
+  (docs "FALSE displays the "invisible char" instead of the actual text (password mode)")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -23289,7 +23289,7 @@
 (define-property invisible-char
   (of-object "GtkSpinButton")
   (prop-type "GParamUnichar")
-  (docs "The character to use when masking entry contents (in password mode)")
+  (docs "The character to use when masking entry contents (in "password mode")")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -23599,7 +23599,7 @@
 (define-property no-show-all
   (of-object "GtkStatusbar")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -23840,7 +23840,7 @@
 (define-property no-show-all
   (of-object "GtkTable")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -24085,7 +24085,7 @@
 (define-property no-show-all
   (of-object "GtkTearoffMenuItem")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -24320,7 +24320,7 @@
 (define-property no-show-all
   (of-object "GtkTipsQuery")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -24383,7 +24383,7 @@
 (define-property use-markup
   (of-object "GtkTipsQuery")
   (prop-type "GParamBoolean")
-  (docs "The text of the label includes XML markup. See pango_parse_markup")
+  (docs "The text of the label includes XML markup. See pango_parse_markup()")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -24724,7 +24724,7 @@
 (define-property no-show-all
   (of-object "GtkToggleButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -24841,7 +24841,7 @@
 (define-property inconsistent
   (of-object "GtkToggleButton")
   (prop-type "GParamBoolean")
-  (docs "If the toggle button is in an in-between state")
+  (docs "If the toggle button is in an "in between" state")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -25070,7 +25070,7 @@
 (define-property no-show-all
   (of-object "GtkToolbar")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -25329,7 +25329,7 @@
 (define-property no-show-all
   (of-object "GtkToolItem")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -25562,7 +25562,7 @@
 (define-property no-show-all
   (of-object "GtkToolButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -25736,7 +25736,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("const-GtkTextIter*" "p0")
+    '("GtkTextIter*" "p0")
     '("GtkTextMark*" "p1")
   )
 )
@@ -25756,8 +25756,8 @@
   (when "last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("const-GtkTextIter*" "p1")
-    '("const-GtkTextIter*" "p2")
+    '("GtkTextIter*" "p1")
+    '("GtkTextIter*" "p2")
   )
 )
 
@@ -25767,8 +25767,8 @@
   (when "last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("const-GtkTextIter*" "p1")
-    '("const-GtkTextIter*" "p2")
+    '("GtkTextIter*" "p1")
+    '("GtkTextIter*" "p2")
   )
 )
 
@@ -25813,7 +25813,7 @@
   (parameters
     '("GObject*" "p0")
     '("GdkEvent*" "p1")
-    '("const-GtkTextIter*" "p2")
+    '("GtkTextIter*" "p2")
   )
 )
 
@@ -25883,7 +25883,7 @@
 (define-property font
   (of-object "GtkTextTag")
   (prop-type "GParamString")
-  (docs "Font description as a string, e.g. Sans Italic 12")
+  (docs "Font description as a string, e.g. "Sans Italic 12"")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -26081,7 +26081,7 @@
 (define-property rise
   (of-object "GtkTextTag")
   (prop-type "GParamInt")
-  (docs "Offset of text above the baseline (below the baseline if rise is negative) in pixels")
+  (docs "Offset of text above the baseline (below the baseline if rise is negative) in Pango units")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -26713,7 +26713,7 @@
 (define-property no-show-all
   (of-object "GtkTextView")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -27179,7 +27179,7 @@
 (define-property no-show-all
   (of-object "GtkToggleToolButton")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -27642,7 +27642,7 @@
 (define-property no-show-all
   (of-object "GtkTreeView")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -28203,7 +28203,7 @@
 (define-property no-show-all
   (of-object "GtkViewport")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -28315,7 +28315,7 @@
   (return-type "void")
   (when "first")
   (parameters
-    '("GtkAllocation*" "p0")
+    '("GdkRectangle*" "p0")
   )
 )
 
@@ -28378,7 +28378,7 @@
   (return-type "void")
   (when "first")
   (parameters
-    '("GParamSpec*" "p0")
+    '("GParam" "p0")
   )
 )
 
@@ -28429,7 +28429,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventButton*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28438,7 +28438,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventButton*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28447,7 +28447,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventScroll*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28456,7 +28456,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventMotion*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28465,7 +28465,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventAny*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28474,7 +28474,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventAny*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28483,7 +28483,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventExpose*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28492,7 +28492,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventKey*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28501,7 +28501,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventKey*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28510,7 +28510,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventCrossing*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28519,7 +28519,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventCrossing*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28528,7 +28528,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventConfigure*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28537,7 +28537,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventFocus*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28546,7 +28546,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventFocus*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28555,7 +28555,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventAny*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28564,7 +28564,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventAny*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28573,7 +28573,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventProperty*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28582,7 +28582,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventSelection*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28591,7 +28591,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventSelection*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28600,7 +28600,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventSelection*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28630,7 +28630,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventProximity*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28639,7 +28639,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventProximity*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28735,7 +28735,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventVisibility*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28744,7 +28744,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventClient*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28753,7 +28753,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventAny*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28762,7 +28762,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEventWindowState*" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
@@ -28979,7 +28979,7 @@
 (define-property no-show-all
   (of-object "GtkWidget")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -29197,7 +29197,7 @@
 (define-property no-show-all
   (of-object "GtkWindow")
   (prop-type "GParamBoolean")
-  (docs "Whether gtk_widget_show_all should not affect this widget")
+  (docs "Whether gtk_widget_show_all() should not affect this widget")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -29445,3 +29445,4 @@
   (writable #f)
   (construct-only #f)
 )
+
