--- tools/gen_scripts/../../gtk/src/gtk_signals.defs.orig	2022-07-22 16:47:55.780698020 +0200
+++ tools/gen_scripts/../../gtk/src/gtk_signals.defs	2022-07-23 10:38:31.807933062 +0200
@@ -2312,11 +2312,11 @@
 (define-signal color-activated
   (of-object "GtkColorChooser")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GdkRGBA*" "p0")
+    '("const-GdkRGBA*" "p0")
   )
 )
 
 (define-property rgba
   (of-object "GtkColorChooser")
@@ -2560,11 +2560,11 @@
   (flags "Run Last, Action")
 )
 
 (define-signal format-entry-text
   (of-object "GtkComboBox")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (flags "Run Last")
   (parameters
     '("const-gchar*" "p0")
   )
 )
@@ -2894,11 +2894,11 @@
   (of-object "GtkCssProvider")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkCssSection*" "p0")
-    '("GError*" "p1")
+    '("const-GError*" "p1")
   )
 )
 
 ;; From GtkDialog
 
@@ -3254,11 +3254,11 @@
 (define-signal drop
   (of-object "GtkDropTarget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GValue*" "p0")
+    '("const-GValue*" "p0")
     '("gdouble" "p1")
     '("gdouble" "p2")
   )
 )
 
@@ -4262,11 +4262,11 @@
 (define-signal event
   (of-object "GtkEventControllerLegacy")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
 ;; From GtkEventControllerMotion
 
@@ -8295,11 +8295,11 @@
   )
 )
 
 (define-signal create-custom-widget
   (of-object "GtkPrintOperation")
-  (return-type "GObject*")
+  (return-type "GtkWidget*")
   (flags "Run Last")
 )
 
 (define-signal update-custom-widget
   (of-object "GtkPrintOperation")
@@ -10219,38 +10219,38 @@
 (define-signal setup
   (of-object "GtkSignalListItemFactory")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GObject*" "p0")
+    '("GtkListItem*" "p0")
   )
 )
 
 (define-signal bind
   (of-object "GtkSignalListItemFactory")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GObject*" "p0")
+    '("GtkListItem*" "p0")
   )
 )
 
 (define-signal unbind
   (of-object "GtkSignalListItemFactory")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GObject*" "p0")
+    '("GtkListItem*" "p0")
   )
 )
 
 (define-signal teardown
   (of-object "GtkSignalListItemFactory")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GObject*" "p0")
+    '("GtkListItem*" "p0")
   )
 )
 
 ;; From GtkSingleSelection
 
@@ -10485,11 +10485,11 @@
 (define-signal input
   (of-object "GtkSpinButton")
   (return-type "gint")
   (flags "Run Last")
   (parameters
-    '("gpointer" "p0")
+    '("double*" "p0")
   )
 )
 
 (define-signal output
   (of-object "GtkSpinButton")
@@ -11234,11 +11234,11 @@
 (define-signal mark-set
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
-    '("GtkTextIter*" "p0")
+    '("const-GtkTextIter*" "p0")
     '("GtkTextMark*" "p1")
   )
 )
 
 (define-signal mark-deleted
@@ -11254,23 +11254,23 @@
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal remove-tag
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal begin-user-action
   (of-object "GtkTextBuffer")
@@ -12751,11 +12751,11 @@
   (return-type "void")
   (flags "Run First")
   (parameters
     '("GtkTreePath*" "p0")
     '("GtkTreeIter*" "p1")
-    '("gpointer" "p2")
+    '("gint*" "p2")
   )
 )
 
 ;; From GtkTreeListModel
 
