--- ./../../gtk/src/gtk_signals.defs.orig	2018-11-19 12:56:33.963564365 +0100
+++ ./../../gtk/src/gtk_signals.defs	2018-11-19 12:56:34.019564544 +0100
@@ -2283,11 +2283,11 @@
 (define-signal color-activated
   (of-object "GtkColorChooser")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GdkRGBA*" "p0")
+    '("const-GdkRGBA*" "p0")
   )
 )
 
 (define-property rgba
   (of-object "GtkColorChooser")
@@ -2349,11 +2349,11 @@
   (flags "Run Last, Action")
 )
 
 (define-signal format-entry-text
   (of-object "GtkComboBox")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (flags "Run Last")
   (parameters
     '("const-gchar*" "p0")
   )
 )
@@ -2577,11 +2577,11 @@
   (of-object "GtkCssProvider")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkCssSection*" "p0")
-    '("GError*" "p1")
+    '("const-GError*" "p1")
   )
 )
 
 ;; From GtkDialog
 
@@ -6049,11 +6049,11 @@
 (define-signal toggle-size-request
   (of-object "GtkMenuItem")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("gpointer" "p0")
+    '("int*" "p0")
   )
 )
 
 (define-signal toggle-size-allocate
   (of-object "GtkMenuItem")
@@ -7068,11 +7068,11 @@
   )
 )
 
 (define-signal create-custom-widget
   (of-object "GtkPrintOperation")
-  (return-type "GObject*")
+  (return-type "GtkWidget*")
   (flags "Run Last")
 )
 
 (define-signal update-custom-widget
   (of-object "GtkPrintOperation")
@@ -7682,11 +7682,11 @@
 
 ;; From GtkScale
 
 (define-signal format-value
   (of-object "GtkScale")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (flags "Run Last")
   (parameters
     '("gdouble" "p0")
   )
 )
@@ -8864,11 +8864,11 @@
 (define-signal input
   (of-object "GtkSpinButton")
   (return-type "gint")
   (flags "Run Last")
   (parameters
-    '("gpointer" "p0")
+    '("double*" "p0")
   )
 )
 
 (define-signal output
   (of-object "GtkSpinButton")
@@ -9433,11 +9433,11 @@
 (define-signal mark-set
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
-    '("GtkTextIter*" "p0")
+    '("const-GtkTextIter*" "p0")
     '("GtkTextMark*" "p1")
   )
 )
 
 (define-signal mark-deleted
@@ -9453,23 +9453,23 @@
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal remove-tag
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal begin-user-action
   (of-object "GtkTextBuffer")
@@ -10727,11 +10727,11 @@
   (return-type "void")
   (flags "Run First")
   (parameters
     '("GtkTreePath*" "p0")
     '("GtkTreeIter*" "p1")
-    '("gpointer" "p2")
+    '("gint*" "p2")
   )
 )
 
 ;; From GtkTreeSelection
 
@@ -11445,11 +11445,11 @@
   (of-object "GtkWidget")
   (return-type "void")
   (flags "Run First, No Recurse, No Hooks")
   (detailed #t)
   (parameters
-    '("GParam" "p0")
+    '("GParamSpec*" "p0")
   )
 )
 
 (define-signal mnemonic-activate
   (of-object "GtkWidget")
