--- ./../../gtk/src/gtk_signals.defs.orig	2018-04-11 18:14:10.215329071 +0200
+++ ./../../gtk/src/gtk_signals.defs	2018-04-11 18:27:32.206901498 +0200
@@ -2267,11 +2267,11 @@
 (define-signal color-activated
   (of-object "GtkColorChooser")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GdkRGBA*" "p0")
+    '("const-GdkRGBA*" "p0")
   )
 )
 
 (define-property rgba
   (of-object "GtkColorChooser")
@@ -2333,11 +2333,11 @@
   (flags "Run Last, Action")
 )
 
 (define-signal format-entry-text
   (of-object "GtkComboBox")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (flags "Run Last")
   (parameters
     '("const-gchar*" "p0")
   )
 )
@@ -2561,11 +2561,11 @@
   (of-object "GtkCssProvider")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkCssSection*" "p0")
-    '("GError*" "p1")
+    '("const-GError*" "p1")
   )
 )
 
 ;; From GtkDialog
 
@@ -2726,21 +2726,21 @@
   (of-object "GtkEntry")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkEntryIconPosition" "p0")
-    '("GdkEvent*" "p1")
+    '("const-GdkEventButton*" "p1")
   )
 )
 
 (define-signal icon-release
   (of-object "GtkEntry")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkEntryIconPosition" "p0")
-    '("GdkEvent*" "p1")
+    '("const-GdkEventButton*" "p1")
   )
 )
 
 (define-signal preedit-changed
   (of-object "GtkEntry")
@@ -5718,11 +5718,11 @@
 (define-signal toggle-size-request
   (of-object "GtkMenuItem")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("gpointer" "p0")
+    '("int*" "p0")
   )
 )
 
 (define-signal toggle-size-allocate
   (of-object "GtkMenuItem")
@@ -6677,11 +6677,11 @@
   )
 )
 
 (define-signal create-custom-widget
   (of-object "GtkPrintOperation")
-  (return-type "GObject*")
+  (return-type "GtkWidget*")
   (flags "Run Last")
 )
 
 (define-signal update-custom-widget
   (of-object "GtkPrintOperation")
@@ -7291,11 +7291,11 @@
 
 ;; From GtkScale
 
 (define-signal format-value
   (of-object "GtkScale")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (flags "Run Last")
   (parameters
     '("gdouble" "p0")
   )
 )
@@ -8473,11 +8473,11 @@
 (define-signal input
   (of-object "GtkSpinButton")
   (return-type "gint")
   (flags "Run Last")
   (parameters
-    '("gpointer" "p0")
+    '("double*" "p0")
   )
 )
 
 (define-signal output
   (of-object "GtkSpinButton")
@@ -9061,11 +9061,11 @@
 (define-signal mark-set
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
-    '("GtkTextIter*" "p0")
+    '("const-GtkTextIter*" "p0")
     '("GtkTextMark*" "p1")
   )
 )
 
 (define-signal mark-deleted
@@ -9081,23 +9081,23 @@
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal remove-tag
   (of-object "GtkTextBuffer")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal begin-user-action
   (of-object "GtkTextBuffer")
@@ -10355,11 +10355,11 @@
   (return-type "void")
   (flags "Run First")
   (parameters
     '("GtkTreePath*" "p0")
     '("GtkTreeIter*" "p1")
-    '("gpointer" "p2")
+    '("gint*" "p2")
   )
 )
 
 ;; From GtkTreeSelection
 
@@ -10979,11 +10979,11 @@
 (define-signal size-allocate
   (of-object "GtkWidget")
   (return-type "void")
   (flags "Run First")
   (parameters
-    '("GdkRectangle*" "p0")
+    '("const-GdkRectangle*" "p0")
     '("gint" "p1")
   )
 )
 
 (define-signal state-flags-changed
@@ -11032,11 +11032,11 @@
   (of-object "GtkWidget")
   (return-type "void")
   (flags "Run First, No Recurse, No Hooks")
   (detailed #t)
   (parameters
-    '("GParam" "p0")
+    '("GParamSpec*" "p0")
   )
 )
 
 (define-signal mnemonic-activate
   (of-object "GtkWidget")
@@ -11093,20 +11093,20 @@
 (define-signal key-press-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventKey*" "p0")
   )
 )
 
 (define-signal key-release-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventKey*" "p0")
   )
 )
 
 (define-signal drag-leave
   (of-object "GtkWidget")
