Index: gtk_signals.defs
===================================================================
--- gtk_signals.defs	(revision 1002)
+++ gtk_signals.defs	(working copy)
@@ -24,7 +24,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("gpointer" "p0")
+    '("gint*" "p0")
   )
 )
 
@@ -558,6 +558,12 @@
   (when "last")
 )
 
+(define-signal monitors-changed
+  (of-object "GdkScreen")
+  (return-type "void")
+  (when "last")
+)
+
 (define-property font-options
   (of-object "GdkScreen")
   (prop-type "GParamPointer")
@@ -780,6 +786,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkAboutDialog")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkAboutDialog")
   (prop-type "GParamUInt")
@@ -1218,6 +1233,24 @@
   )
 )
 
+(define-property is-locked
+  (of-object "GtkAccelGroup")
+  (prop-type "GParamBoolean")
+  (docs "Is the accel group locked")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
+(define-property modifier-mask
+  (of-object "GtkAccelGroup")
+  (prop-type "GParamFlags")
+  (docs "Modifier Mask")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 ;; From GtkAccelLabel
 
 (define-property user-data
@@ -1418,6 +1451,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkAccelLabel")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property xalign
   (of-object "GtkAccelLabel")
   (prop-type "GParamFloat")
@@ -1837,13 +1879,13 @@
 
 ;; From GtkAdjustment
 
-(define-signal changed
+(define-signal value-changed
   (of-object "GtkAdjustment")
   (return-type "void")
   (when "first")
 )
 
-(define-signal value-changed
+(define-signal changed
   (of-object "GtkAdjustment")
   (return-type "void")
   (when "first")
@@ -2112,6 +2154,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkAlignment")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkAlignment")
   (prop-type "GParamUInt")
@@ -2411,6 +2462,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkArrow")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property xalign
   (of-object "GtkArrow")
   (prop-type "GParamFloat")
@@ -2665,6 +2725,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkAspectFrame")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkAspectFrame")
   (prop-type "GParamUInt")
@@ -3009,6 +3078,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkAssistant")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkAssistant")
   (prop-type "GParamUInt")
@@ -3488,6 +3566,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkBin")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkBin")
   (prop-type "GParamUInt")
@@ -3715,6 +3802,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkBox")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkBox")
   (prop-type "GParamUInt")
@@ -3760,39 +3856,50 @@
   (construct-only #f)
 )
 
+;; From GtkBuilder
+
+(define-property translation-domain
+  (of-object "GtkBuilder")
+  (prop-type "GParamString")
+  (docs "The translation domain used by gettext")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkButton
 
-(define-signal clicked
+(define-signal pressed
   (of-object "GtkButton")
   (return-type "void")
   (when "first")
 )
 
-(define-signal activate
+(define-signal released
   (of-object "GtkButton")
   (return-type "void")
   (when "first")
 )
 
-(define-signal pressed
+(define-signal clicked
   (of-object "GtkButton")
   (return-type "void")
   (when "first")
 )
 
-(define-signal released
+(define-signal enter
   (of-object "GtkButton")
   (return-type "void")
   (when "first")
 )
 
-(define-signal enter
+(define-signal leave
   (of-object "GtkButton")
   (return-type "void")
   (when "first")
 )
 
-(define-signal leave
+(define-signal activate
   (of-object "GtkButton")
   (return-type "void")
   (when "first")
@@ -3996,6 +4103,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkButton")
   (prop-type "GParamUInt")
@@ -4304,6 +4420,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkButtonBox")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkButtonBox")
   (prop-type "GParamUInt")
@@ -4600,6 +4725,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkCalendar")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property year
   (of-object "GtkCalendar")
   (prop-type "GParamInt")
@@ -4663,6 +4797,33 @@
   (construct-only #f)
 )
 
+(define-property show-details
+  (of-object "GtkCalendar")
+  (prop-type "GParamBoolean")
+  (docs "If TRUE, details are shown")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property detail-width-chars
+  (of-object "GtkCalendar")
+  (prop-type "GParamInt")
+  (docs "Details width in characters")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property detail-height-rows
+  (of-object "GtkCalendar")
+  (prop-type "GParamInt")
+  (docs "Details height in rows")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkCheckButton
 
 (define-property user-data
@@ -4863,6 +5024,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkCheckButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkCheckButton")
   (prop-type "GParamUInt")
@@ -5204,6 +5374,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkCheckMenuItem")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkCheckMenuItem")
   (prop-type "GParamUInt")
@@ -5231,6 +5410,15 @@
   (construct-only #f)
 )
 
+(define-property right-justified
+  (of-object "GtkCheckMenuItem")
+  (prop-type "GParamBoolean")
+  (docs "Sets whether the menu item appears justified at the right side of a menu bar")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property submenu
   (of-object "GtkCheckMenuItem")
   (prop-type "GParamObject")
@@ -5240,6 +5428,15 @@
   (construct-only #f)
 )
 
+(define-property accel-path
+  (of-object "GtkCheckMenuItem")
+  (prop-type "GParamString")
+  (docs "Sets the accelerator path of the menu item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property active
   (of-object "GtkCheckMenuItem")
   (prop-type "GParamBoolean")
@@ -5607,6 +5804,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkCList")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkCList")
   (prop-type "GParamUInt")
@@ -5861,6 +6067,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRenderer")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 ;; From GtkCellRendererAccel
 
 (define-signal accel-edited
@@ -6019,6 +6234,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRendererAccel")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property text
   (of-object "GtkCellRendererAccel")
   (prop-type "GParamString")
@@ -6453,6 +6677,16 @@
 
 ;; From GtkCellRendererCombo
 
+(define-signal changed
+  (of-object "GtkCellRendererCombo")
+  (return-type "void")
+  (when "last")
+  (parameters
+    '("const-gchar*" "p0")
+    '("GtkTreeIter*" "p1")
+  )
+)
+
 (define-property user-data
   (of-object "GtkCellRendererCombo")
   (prop-type "GParamPointer")
@@ -6588,6 +6822,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRendererCombo")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property text
   (of-object "GtkCellRendererCombo")
   (prop-type "GParamString")
@@ -7148,6 +7391,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRendererProgress")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property value
   (of-object "GtkCellRendererProgress")
   (prop-type "GParamInt")
@@ -7349,6 +7601,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRendererText")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property text
   (of-object "GtkCellRendererText")
   (prop-type "GParamString")
@@ -7891,6 +8152,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRendererToggle")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property activatable
   (of-object "GtkCellRendererToggle")
   (prop-type "GParamBoolean")
@@ -8073,6 +8343,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRendererPixbuf")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property pixbuf
   (of-object "GtkCellRendererPixbuf")
   (prop-type "GParamObject")
@@ -8145,6 +8424,15 @@
   (construct-only #f)
 )
 
+(define-property gicon
+  (of-object "GtkCellRendererPixbuf")
+  (prop-type "GParamObject")
+  (docs "The GIcon being displayed")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkCellRendererProgress
 
 (define-property user-data
@@ -8282,6 +8570,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRendererProgress")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property value
   (of-object "GtkCellRendererProgress")
   (prop-type "GParamInt")
@@ -8473,6 +8770,15 @@
   (construct-only #f)
 )
 
+(define-property editing
+  (of-object "GtkCellRendererSpin")
+  (prop-type "GParamBoolean")
+  (docs "Whether the cell renderer is currently in editing mode")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property text
   (of-object "GtkCellRendererSpin")
   (prop-type "GParamString")
@@ -9113,6 +9419,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkColorButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkColorButton")
   (prop-type "GParamUInt")
@@ -9463,6 +9778,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkColorSelection")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkColorSelection")
   (prop-type "GParamUInt")
@@ -9744,6 +10068,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkCombo")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkCombo")
   (prop-type "GParamUInt")
@@ -9836,31 +10169,31 @@
 
 ;; From GtkComboBox
 
-(define-signal changed
+(define-signal popup
   (of-object "GtkComboBox")
   (return-type "void")
   (when "last")
 )
 
-(define-signal move-active
+(define-signal popdown
   (of-object "GtkComboBox")
-  (return-type "void")
+  (return-type "gboolean")
   (when "last")
-  (parameters
-    '("GtkScrollType" "p0")
-  )
 )
 
-(define-signal popup
+(define-signal changed
   (of-object "GtkComboBox")
   (return-type "void")
   (when "last")
 )
 
-(define-signal popdown
+(define-signal move-active
   (of-object "GtkComboBox")
-  (return-type "gboolean")
+  (return-type "void")
   (when "last")
+  (parameters
+    '("GtkScrollType" "p0")
+  )
 )
 
 (define-property user-data
@@ -10061,6 +10394,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkComboBox")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkComboBox")
   (prop-type "GParamUInt")
@@ -10411,6 +10753,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkContainer")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkContainer")
   (prop-type "GParamUInt")
@@ -10696,6 +11047,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkCTree")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkCTree")
   (prop-type "GParamUInt")
@@ -11049,6 +11409,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkCellView")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property background
   (of-object "GtkCellView")
   (prop-type "GParamString")
@@ -11291,6 +11660,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkCurve")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property curve-type
   (of-object "GtkCurve")
   (prop-type "GParamEnum")
@@ -11551,6 +11929,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkDialog")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkDialog")
   (prop-type "GParamUInt")
@@ -12039,8 +12426,23 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkDrawingArea")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 ;; From GtkEntry
 
+(define-signal activate
+  (of-object "GtkEntry")
+  (return-type "void")
+  (when "last")
+)
+
 (define-signal move-cursor
   (of-object "GtkEntry")
   (return-type "void")
@@ -12052,12 +12454,6 @@
   )
 )
 
-(define-signal activate
-  (of-object "GtkEntry")
-  (return-type "void")
-  (when "last")
-)
-
 (define-signal insert-at-cursor
   (of-object "GtkEntry")
   (return-type "void")
@@ -12314,6 +12710,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkEntry")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property cursor-position
   (of-object "GtkEntry")
   (prop-type "GParamInt")
@@ -12449,6 +12854,24 @@
   (construct-only #f)
 )
 
+(define-property overwrite-mode
+  (of-object "GtkEntry")
+  (prop-type "GParamBoolean")
+  (docs "Whether new text overwrites existing text")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property text-length
+  (of-object "GtkEntry")
+  (prop-type "GParamUInt")
+  (docs "Length of the text currently in the entry")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 ;; From GtkEntryCompletion
 
 (define-signal insert-prefix
@@ -12761,6 +13184,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkEventBox")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkEventBox")
   (prop-type "GParamUInt")
@@ -13012,6 +13444,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkExpander")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkExpander")
   (prop-type "GParamUInt")
@@ -13112,10 +13553,10 @@
   (when "first")
 )
 
-(define-property action
+(define-property preview-widget
   (of-object "GtkFileChooserButton")
-  (prop-type "GParamEnum")
-  (docs "The type of operation that the file selector is performing")
+  (prop-type "GParamObject")
+  (docs "Application supplied widget for custom previews.")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -13130,82 +13571,82 @@
   (construct-only #f)
 )
 
-(define-property extra-widget
+(define-property show-hidden
   (of-object "GtkFileChooserButton")
-  (prop-type "GParamObject")
-  (docs "Application supplied widget for extra options.")
+  (prop-type "GParamBoolean")
+  (docs "Whether the hidden files and folders should be displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property file-system-backend
+(define-property extra-widget
   (of-object "GtkFileChooserButton")
-  (prop-type "GParamString")
-  (docs "Name of file system backend to use")
-  (readable #f)
+  (prop-type "GParamObject")
+  (docs "Application supplied widget for extra options.")
+  (readable #t)
   (writable #t)
-  (construct-only #t)
+  (construct-only #f)
 )
 
-(define-property select-multiple
+(define-property filter
   (of-object "GtkFileChooserButton")
-  (prop-type "GParamBoolean")
-  (docs "Whether to allow multiple files to be selected")
+  (prop-type "GParamObject")
+  (docs "The current filter for selecting which files are displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property show-hidden
+(define-property local-only
   (of-object "GtkFileChooserButton")
   (prop-type "GParamBoolean")
-  (docs "Whether the hidden files and folders should be displayed")
+  (docs "Whether the selected file(s) should be limited to local file: URLs")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property preview-widget
+(define-property action
   (of-object "GtkFileChooserButton")
-  (prop-type "GParamObject")
-  (docs "Application supplied widget for custom previews.")
+  (prop-type "GParamEnum")
+  (docs "The type of operation that the file selector is performing")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property local-only
+(define-property file-system-backend
   (of-object "GtkFileChooserButton")
-  (prop-type "GParamBoolean")
-  (docs "Whether the selected file(s) should be limited to local file: URLs")
-  (readable #t)
+  (prop-type "GParamString")
+  (docs "Name of file system backend to use")
+  (readable #f)
   (writable #t)
-  (construct-only #f)
+  (construct-only #t)
 )
 
-(define-property preview-widget-active
+(define-property select-multiple
   (of-object "GtkFileChooserButton")
   (prop-type "GParamBoolean")
-  (docs "Whether the application supplied widget for custom previews should be shown.")
+  (docs "Whether to allow multiple files to be selected")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property filter
+(define-property do-overwrite-confirmation
   (of-object "GtkFileChooserButton")
-  (prop-type "GParamObject")
-  (docs "The current filter for selecting which files are displayed")
+  (prop-type "GParamBoolean")
+  (docs "Whether a file chooser in save mode will present an overwrite confirmation dialog if necessary.")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property do-overwrite-confirmation
+(define-property preview-widget-active
   (of-object "GtkFileChooserButton")
   (prop-type "GParamBoolean")
-  (docs "Whether a file chooser in save mode will present an overwrite confirmation dialog if necessary.")
+  (docs "Whether the application supplied widget for custom previews should be shown.")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -13409,6 +13850,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFileChooserButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFileChooserButton")
   (prop-type "GParamUInt")
@@ -13492,10 +13942,10 @@
 
 ;; From GtkFileChooserDialog
 
-(define-property action
+(define-property preview-widget
   (of-object "GtkFileChooserDialog")
-  (prop-type "GParamEnum")
-  (docs "The type of operation that the file selector is performing")
+  (prop-type "GParamObject")
+  (docs "Application supplied widget for custom previews.")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -13510,82 +13960,82 @@
   (construct-only #f)
 )
 
-(define-property extra-widget
+(define-property show-hidden
   (of-object "GtkFileChooserDialog")
-  (prop-type "GParamObject")
-  (docs "Application supplied widget for extra options.")
+  (prop-type "GParamBoolean")
+  (docs "Whether the hidden files and folders should be displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property file-system-backend
+(define-property extra-widget
   (of-object "GtkFileChooserDialog")
-  (prop-type "GParamString")
-  (docs "Name of file system backend to use")
-  (readable #f)
+  (prop-type "GParamObject")
+  (docs "Application supplied widget for extra options.")
+  (readable #t)
   (writable #t)
-  (construct-only #t)
+  (construct-only #f)
 )
 
-(define-property select-multiple
+(define-property filter
   (of-object "GtkFileChooserDialog")
-  (prop-type "GParamBoolean")
-  (docs "Whether to allow multiple files to be selected")
+  (prop-type "GParamObject")
+  (docs "The current filter for selecting which files are displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property show-hidden
+(define-property local-only
   (of-object "GtkFileChooserDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether the hidden files and folders should be displayed")
+  (docs "Whether the selected file(s) should be limited to local file: URLs")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property preview-widget
+(define-property action
   (of-object "GtkFileChooserDialog")
-  (prop-type "GParamObject")
-  (docs "Application supplied widget for custom previews.")
+  (prop-type "GParamEnum")
+  (docs "The type of operation that the file selector is performing")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property local-only
+(define-property file-system-backend
   (of-object "GtkFileChooserDialog")
-  (prop-type "GParamBoolean")
-  (docs "Whether the selected file(s) should be limited to local file: URLs")
-  (readable #t)
+  (prop-type "GParamString")
+  (docs "Name of file system backend to use")
+  (readable #f)
   (writable #t)
-  (construct-only #f)
+  (construct-only #t)
 )
 
-(define-property preview-widget-active
+(define-property select-multiple
   (of-object "GtkFileChooserDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether the application supplied widget for custom previews should be shown.")
+  (docs "Whether to allow multiple files to be selected")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property filter
+(define-property do-overwrite-confirmation
   (of-object "GtkFileChooserDialog")
-  (prop-type "GParamObject")
-  (docs "The current filter for selecting which files are displayed")
+  (prop-type "GParamBoolean")
+  (docs "Whether a file chooser in save mode will present an overwrite confirmation dialog if necessary.")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property do-overwrite-confirmation
+(define-property preview-widget-active
   (of-object "GtkFileChooserDialog")
   (prop-type "GParamBoolean")
-  (docs "Whether a file chooser in save mode will present an overwrite confirmation dialog if necessary.")
+  (docs "Whether the application supplied widget for custom previews should be shown.")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -13789,6 +14239,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFileChooserDialog")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFileChooserDialog")
   (prop-type "GParamUInt")
@@ -14079,10 +14538,10 @@
 
 ;; From GtkFileChooserWidget
 
-(define-property action
+(define-property preview-widget
   (of-object "GtkFileChooserWidget")
-  (prop-type "GParamEnum")
-  (docs "The type of operation that the file selector is performing")
+  (prop-type "GParamObject")
+  (docs "Application supplied widget for custom previews.")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -14097,82 +14556,82 @@
   (construct-only #f)
 )
 
-(define-property extra-widget
+(define-property show-hidden
   (of-object "GtkFileChooserWidget")
-  (prop-type "GParamObject")
-  (docs "Application supplied widget for extra options.")
+  (prop-type "GParamBoolean")
+  (docs "Whether the hidden files and folders should be displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property file-system-backend
+(define-property extra-widget
   (of-object "GtkFileChooserWidget")
-  (prop-type "GParamString")
-  (docs "Name of file system backend to use")
-  (readable #f)
+  (prop-type "GParamObject")
+  (docs "Application supplied widget for extra options.")
+  (readable #t)
   (writable #t)
-  (construct-only #t)
+  (construct-only #f)
 )
 
-(define-property select-multiple
+(define-property filter
   (of-object "GtkFileChooserWidget")
-  (prop-type "GParamBoolean")
-  (docs "Whether to allow multiple files to be selected")
+  (prop-type "GParamObject")
+  (docs "The current filter for selecting which files are displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property show-hidden
+(define-property local-only
   (of-object "GtkFileChooserWidget")
   (prop-type "GParamBoolean")
-  (docs "Whether the hidden files and folders should be displayed")
+  (docs "Whether the selected file(s) should be limited to local file: URLs")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property preview-widget
+(define-property action
   (of-object "GtkFileChooserWidget")
-  (prop-type "GParamObject")
-  (docs "Application supplied widget for custom previews.")
+  (prop-type "GParamEnum")
+  (docs "The type of operation that the file selector is performing")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property local-only
+(define-property file-system-backend
   (of-object "GtkFileChooserWidget")
-  (prop-type "GParamBoolean")
-  (docs "Whether the selected file(s) should be limited to local file: URLs")
-  (readable #t)
+  (prop-type "GParamString")
+  (docs "Name of file system backend to use")
+  (readable #f)
   (writable #t)
-  (construct-only #f)
+  (construct-only #t)
 )
 
-(define-property preview-widget-active
+(define-property select-multiple
   (of-object "GtkFileChooserWidget")
   (prop-type "GParamBoolean")
-  (docs "Whether the application supplied widget for custom previews should be shown.")
+  (docs "Whether to allow multiple files to be selected")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property filter
+(define-property do-overwrite-confirmation
   (of-object "GtkFileChooserWidget")
-  (prop-type "GParamObject")
-  (docs "The current filter for selecting which files are displayed")
+  (prop-type "GParamBoolean")
+  (docs "Whether a file chooser in save mode will present an overwrite confirmation dialog if necessary.")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property do-overwrite-confirmation
+(define-property preview-widget-active
   (of-object "GtkFileChooserWidget")
   (prop-type "GParamBoolean")
-  (docs "Whether a file chooser in save mode will present an overwrite confirmation dialog if necessary.")
+  (docs "Whether the application supplied widget for custom previews should be shown.")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -14376,6 +14835,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFileChooserWidget")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFileChooserWidget")
   (prop-type "GParamUInt")
@@ -14621,6 +15089,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFileSelection")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFileSelection")
   (prop-type "GParamUInt")
@@ -15136,6 +15613,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFixed")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFixed")
   (prop-type "GParamUInt")
@@ -15369,6 +15855,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFontButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFontButton")
   (prop-type "GParamUInt")
@@ -15731,6 +16226,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFontSelection")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFontSelection")
   (prop-type "GParamUInt")
@@ -15779,7 +16283,7 @@
 (define-property font-name
   (of-object "GtkFontSelection")
   (prop-type "GParamString")
-  (docs "The X string that represents this font")
+  (docs "The string that represents this font")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -16003,6 +16507,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFontSelectionDialog")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFontSelectionDialog")
   (prop-type "GParamUInt")
@@ -16491,6 +17004,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkFrame")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkFrame")
   (prop-type "GParamUInt")
@@ -16574,7 +17096,7 @@
 
 ;; From GtkHandleBox
 
-(define-signal child-attached
+(define-signal child-detached
   (of-object "GtkHandleBox")
   (return-type "void")
   (when "first")
@@ -16583,7 +17105,7 @@
   )
 )
 
-(define-signal child-detached
+(define-signal child-attached
   (of-object "GtkHandleBox")
   (return-type "void")
   (when "first")
@@ -16790,6 +17312,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkHandleBox")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkHandleBox")
   (prop-type "GParamUInt")
@@ -16862,6 +17393,15 @@
   (construct-only #f)
 )
 
+(define-property child-detached
+  (of-object "GtkHandleBox")
+  (prop-type "GParamBoolean")
+  (docs "A boolean value indicating whether the handlebox's child is attached or detached.")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 ;; From GtkIconTheme
 
 (define-signal changed
@@ -17070,6 +17610,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkImage")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property xalign
   (of-object "GtkImage")
   (prop-type "GParamFloat")
@@ -17214,6 +17763,15 @@
   (construct-only #f)
 )
 
+(define-property gicon
+  (of-object "GtkImage")
+  (prop-type "GParamObject")
+  (docs "The GIcon being displayed")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkInputDialog
 
 (define-signal enable-device
@@ -17432,6 +17990,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkInputDialog")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkInputDialog")
   (prop-type "GParamUInt")
@@ -17920,6 +18487,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkInvisible")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property screen
   (of-object "GtkInvisible")
   (prop-type "GParamObject")
@@ -18147,6 +18723,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkItem")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkItem")
   (prop-type "GParamUInt")
@@ -18439,6 +19024,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkIconView")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkIconView")
   (prop-type "GParamUInt")
@@ -18583,6 +19177,15 @@
   (construct-only #f)
 )
 
+(define-property tooltip-column
+  (of-object "GtkIconView")
+  (prop-type "GParamInt")
+  (docs "The column in the model containing the tooltip texts for the items")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkLabel
 
 (define-signal move-cursor
@@ -18809,6 +19412,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkLabel")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property xalign
   (of-object "GtkLabel")
   (prop-type "GParamFloat")
@@ -19217,6 +19829,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkLayout")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkLayout")
   (prop-type "GParamUInt")
@@ -19480,6 +20101,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkLinkButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkLinkButton")
   (prop-type "GParamUInt")
@@ -19821,6 +20451,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkList")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkList")
   (prop-type "GParamUInt")
@@ -20130,6 +20769,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkListItem")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkListItem")
   (prop-type "GParamUInt")
@@ -20366,6 +21014,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkMenu")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkMenu")
   (prop-type "GParamUInt")
@@ -20402,6 +21059,42 @@
   (construct-only #f)
 )
 
+(define-property active
+  (of-object "GtkMenu")
+  (prop-type "GParamInt")
+  (docs "The currently selected menu item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property accel-group
+  (of-object "GtkMenu")
+  (prop-type "GParamObject")
+  (docs "The accel group holding accelerators for the menu")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property accel-path
+  (of-object "GtkMenu")
+  (prop-type "GParamString")
+  (docs "An accel path used to conveniently construct accel paths of child items")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property attach-widget
+  (of-object "GtkMenu")
+  (prop-type "GParamObject")
+  (docs "The widget the menu is attached to")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property tearoff-state
   (of-object "GtkMenu")
   (prop-type "GParamBoolean")
@@ -20420,6 +21113,15 @@
   (construct-only #f)
 )
 
+(define-property monitor
+  (of-object "GtkMenu")
+  (prop-type "GParamInt")
+  (docs "The monitor the menu will be popped up on")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkMenuBar
 
 (define-property user-data
@@ -20620,6 +21322,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkMenuBar")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkMenuBar")
   (prop-type "GParamUInt")
@@ -20904,6 +21615,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkMenuItem")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkMenuItem")
   (prop-type "GParamUInt")
@@ -20931,6 +21651,15 @@
   (construct-only #f)
 )
 
+(define-property right-justified
+  (of-object "GtkMenuItem")
+  (prop-type "GParamBoolean")
+  (docs "Sets whether the menu item appears justified at the right side of a menu bar")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property submenu
   (of-object "GtkMenuItem")
   (prop-type "GParamObject")
@@ -20940,6 +21669,15 @@
   (construct-only #f)
 )
 
+(define-property accel-path
+  (of-object "GtkMenuItem")
+  (prop-type "GParamString")
+  (docs "Sets the accelerator path of the menu item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkMenuShell
 
 (define-signal deactivate
@@ -21194,6 +21932,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkMenuShell")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkMenuShell")
   (prop-type "GParamUInt")
@@ -21436,6 +22183,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkMenuToolButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkMenuToolButton")
   (prop-type "GParamUInt")
@@ -21753,6 +22509,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkMessageDialog")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkMessageDialog")
   (prop-type "GParamUInt")
@@ -22304,6 +23069,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkMisc")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property xalign
   (of-object "GtkMisc")
   (prop-type "GParamFloat")
@@ -22637,6 +23411,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkNotebook")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkNotebook")
   (prop-type "GParamUInt")
@@ -22995,6 +23778,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkOptionMenu")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkOptionMenu")
   (prop-type "GParamUInt")
@@ -23357,6 +24149,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkPaned")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkPaned")
   (prop-type "GParamUInt")
@@ -23620,6 +24421,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkPixmap")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property xalign
   (of-object "GtkPixmap")
   (prop-type "GParamFloat")
@@ -23862,6 +24672,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkPlug")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkPlug")
   (prop-type "GParamUInt")
@@ -24150,6 +24969,15 @@
   (construct-only #f)
 )
 
+(define-property socket-window
+  (of-object "GtkPlug")
+  (prop-type "GParamObject")
+  (docs "The window of the socket the plug is embedded in")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 ;; From GtkPreview
 
 (define-property user-data
@@ -24350,232 +25178,24 @@
   (construct-only #f)
 )
 
-(define-property expand
+(define-property window
   (of-object "GtkPreview")
-  (prop-type "GParamBoolean")
-  (docs "Whether the preview widget should take up the entire space it is allocated")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-;; From GtkPrintOperation
-
-(define-signal status-changed
-  (of-object "GtkPrintOperation")
-  (return-type "void")
-  (when "last")
-)
-
-(define-signal done
-  (of-object "GtkPrintOperation")
-  (return-type "void")
-  (when "last")
-  (parameters
-    '("GtkPrintOperationResult" "p0")
-  )
-)
-
-(define-signal begin-print
-  (of-object "GtkPrintOperation")
-  (return-type "void")
-  (when "last")
-  (parameters
-    '("GtkPrintContext*" "p0")
-  )
-)
-
-(define-signal paginate
-  (of-object "GtkPrintOperation")
-  (return-type "gboolean")
-  (when "last")
-  (parameters
-    '("GtkPrintContext*" "p0")
-  )
-)
-
-(define-signal request-page-setup
-  (of-object "GtkPrintOperation")
-  (return-type "void")
-  (when "last")
-  (parameters
-    '("GtkPrintContext*" "p0")
-    '("gint" "p1")
-    '("GtkPageSetup*" "p2")
-  )
-)
-
-(define-signal draw-page
-  (of-object "GtkPrintOperation")
-  (return-type "void")
-  (when "last")
-  (parameters
-    '("GtkPrintContext*" "p0")
-    '("gint" "p1")
-  )
-)
-
-(define-signal end-print
-  (of-object "GtkPrintOperation")
-  (return-type "void")
-  (when "last")
-  (parameters
-    '("GtkPrintContext*" "p0")
-  )
-)
-
-(define-signal create-custom-widget
-  (of-object "GtkPrintOperation")
-  (return-type "GtkWidget*")
-  (when "last")
-)
-
-(define-signal custom-widget-apply
-  (of-object "GtkPrintOperation")
-  (return-type "void")
-  (when "last")
-  (parameters
-    '("GtkWidget*" "p0")
-  )
-)
-
-(define-signal preview
-  (of-object "GtkPrintOperation")
-  (return-type "gboolean")
-  (when "last")
-  (parameters
-    '("GtkPrintOperationPreview*" "p0")
-    '("GtkPrintContext*" "p1")
-    '("GtkWindow*" "p2")
-  )
-)
-
-(define-property default-page-setup
-  (of-object "GtkPrintOperation")
   (prop-type "GParamObject")
-  (docs "The GtkPageSetup used by default")
+  (docs "The widget's window if it is realized")
   (readable #t)
-  (writable #t)
+  (writable #f)
   (construct-only #f)
 )
 
-(define-property print-settings
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamObject")
-  (docs "The GtkPrintSettings used for initializing the dialog")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property job-name
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamString")
-  (docs "A string used for identifying the print job.")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property n-pages
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamInt")
-  (docs "The number of pages in the document.")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property current-page
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamInt")
-  (docs "The current page in the document")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property use-full-page
-  (of-object "GtkPrintOperation")
+(define-property expand
+  (of-object "GtkPreview")
   (prop-type "GParamBoolean")
-  (docs "TRUE if the origin of the context should be at the corner of the page and not the corner of the imageable area")
+  (docs "Whether the preview widget should take up the entire space it is allocated")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property track-print-status
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamBoolean")
-  (docs "TRUE if the print operation will continue to report on the print job status after the print data has been sent to the printer or print server.")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property unit
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamEnum")
-  (docs "The unit in which distances can be measured in the context")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property show-progress
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamBoolean")
-  (docs "TRUE if a progress dialog is shown while printing.")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property allow-async
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamBoolean")
-  (docs "TRUE if print process may run asynchronous.")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property export-filename
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamString")
-  (docs "Export filename")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
-(define-property status
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamEnum")
-  (docs "The status of the print operation")
-  (readable #t)
-  (writable #f)
-  (construct-only #f)
-)
-
-(define-property status-string
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamString")
-  (docs "A human-readable description of the status")
-  (readable #t)
-  (writable #f)
-  (construct-only #f)
-)
-
-(define-property custom-tab-label
-  (of-object "GtkPrintOperation")
-  (prop-type "GParamString")
-  (docs "Label for the tab containing custom widgets.")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
 ;; From GtkPrinter
 
 (define-signal details-acquired
@@ -24938,6 +25558,7 @@
   (construct-only #f)
 )
 
+;; From GtkPrintOperationPreview
 
 ;; From GtkPageSetupUnixDialog
 
@@ -25139,6 +25760,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkPageSetupUnixDialog")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkPageSetupUnixDialog")
   (prop-type "GParamUInt")
@@ -25627,6 +26257,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkPrintUnixDialog")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkPrintUnixDialog")
   (prop-type "GParamUInt")
@@ -26151,6 +26790,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkProgress")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property activity-mode
   (of-object "GtkProgress")
   (prop-type "GParamBoolean")
@@ -26387,6 +27035,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkProgressBar")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property activity-mode
   (of-object "GtkProgressBar")
   (prop-type "GParamBoolean")
@@ -26719,6 +27376,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkRadioButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkRadioButton")
   (prop-type "GParamUInt")
@@ -27063,6 +27729,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkRadioToolButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkRadioToolButton")
   (prop-type "GParamUInt")
@@ -27577,6 +28252,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkRadioMenuItem")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkRadioMenuItem")
   (prop-type "GParamUInt")
@@ -27604,6 +28288,15 @@
   (construct-only #f)
 )
 
+(define-property right-justified
+  (of-object "GtkRadioMenuItem")
+  (prop-type "GParamBoolean")
+  (docs "Sets whether the menu item appears justified at the right side of a menu bar")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property submenu
   (of-object "GtkRadioMenuItem")
   (prop-type "GParamObject")
@@ -27613,6 +28306,15 @@
   (construct-only #f)
 )
 
+(define-property accel-path
+  (of-object "GtkRadioMenuItem")
+  (prop-type "GParamString")
+  (docs "Sets the accelerator path of the menu item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property active
   (of-object "GtkRadioMenuItem")
   (prop-type "GParamBoolean")
@@ -27883,6 +28585,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkRange")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property update-policy
   (of-object "GtkRange")
   (prop-type "GParamEnum")
@@ -27955,49 +28666,57 @@
   (construct-only #f)
 )
 
-
 ;; From GtkRecentAction
 
-(define-property sort-type
+(define-property show-icons
   (of-object "GtkRecentAction")
-  (prop-type "GParamEnum")
-  (docs "The sorting order of the items displayed")
+  (prop-type "GParamBoolean")
+  (docs "Whether there should be an icon near the item")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property filter
+(define-property select-multiple
   (of-object "GtkRecentAction")
-  (prop-type "GParamObject")
-  (docs "The current filter for selecting which resources are displayed")
+  (prop-type "GParamBoolean")
+  (docs "Whether to allow multiple items to be selected")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property show-private
+(define-property sort-type
   (of-object "GtkRecentAction")
-  (prop-type "GParamBoolean")
-  (docs "Whether the private items should be displayed")
+  (prop-type "GParamEnum")
+  (docs "The sorting order of the items displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property show-tips
+(define-property limit
   (of-object "GtkRecentAction")
-  (prop-type "GParamBoolean")
-  (docs "Whether there should be a tooltip on the item")
+  (prop-type "GParamInt")
+  (docs "The maximum number of items to be displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property limit
+(define-property recent-manager
   (of-object "GtkRecentAction")
-  (prop-type "GParamInt")
-  (docs "The maximum number of items to be displayed")
+  (prop-type "GParamObject")
+  (docs "The RecentManager object to use")
+  (readable #f)
+  (writable #t)
+  (construct-only #t)
+)
+
+(define-property filter
+  (of-object "GtkRecentAction")
+  (prop-type "GParamObject")
+  (docs "The current filter for selecting which resources are displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -28012,37 +28731,28 @@
   (construct-only #f)
 )
 
-(define-property show-not-found
+(define-property show-private
   (of-object "GtkRecentAction")
   (prop-type "GParamBoolean")
-  (docs "Whether the items pointing to unavailable resources should be displayed")
+  (docs "Whether the private items should be displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property show-icons
+(define-property show-not-found
   (of-object "GtkRecentAction")
   (prop-type "GParamBoolean")
-  (docs "Whether there should be an icon near the item")
+  (docs "Whether the items pointing to unavailable resources should be displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property recent-manager
+(define-property show-tips
   (of-object "GtkRecentAction")
-  (prop-type "GParamObject")
-  (docs "The RecentManager object to use")
-  (readable #f)
-  (writable #t)
-  (construct-only #t)
-)
-
-(define-property select-multiple
-  (of-object "GtkRecentAction")
   (prop-type "GParamBoolean")
-  (docs "Whether to allow multiple items to be selected")
+  (docs "Whether there should be a tooltip on the item")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -28197,19 +28907,35 @@
   (when "last")
 )
 
-;; From GtkRecentChooserMenu
+(define-property show-icons
+  (of-object "GtkRecentChooser")
+  (prop-type "GParamBoolean")
+  (docs "Whether there should be an icon near the item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
 
-(define-property show-tips
-  (of-object "GtkRecentChooserMenu")
+(define-property select-multiple
+  (of-object "GtkRecentChooser")
   (prop-type "GParamBoolean")
-  (docs "Whether there should be a tooltip on the item")
+  (docs "Whether to allow multiple items to be selected")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
+(define-property sort-type
+  (of-object "GtkRecentChooser")
+  (prop-type "GParamEnum")
+  (docs "The sorting order of the items displayed")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property limit
-  (of-object "GtkRecentChooserMenu")
+  (of-object "GtkRecentChooser")
   (prop-type "GParamInt")
   (docs "The maximum number of items to be displayed")
   (readable #t)
@@ -28217,8 +28943,17 @@
   (construct-only #f)
 )
 
+(define-property recent-manager
+  (of-object "GtkRecentChooser")
+  (prop-type "GParamObject")
+  (docs "The RecentManager object to use")
+  (readable #f)
+  (writable #t)
+  (construct-only #t)
+)
+
 (define-property filter
-  (of-object "GtkRecentChooserMenu")
+  (of-object "GtkRecentChooser")
   (prop-type "GParamObject")
   (docs "The current filter for selecting which resources are displayed")
   (readable #t)
@@ -28226,28 +28961,57 @@
   (construct-only #f)
 )
 
-(define-property recent-manager
-  (of-object "GtkRecentChooserMenu")
-  (prop-type "GParamObject")
-  (docs "The RecentManager object to use")
-  (readable #f)
+(define-property local-only
+  (of-object "GtkRecentChooser")
+  (prop-type "GParamBoolean")
+  (docs "Whether the selected resource(s) should be limited to local file: URIs")
+  (readable #t)
   (writable #t)
-  (construct-only #t)
+  (construct-only #f)
 )
 
-(define-property select-multiple
+(define-property show-private
+  (of-object "GtkRecentChooser")
+  (prop-type "GParamBoolean")
+  (docs "Whether the private items should be displayed")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property show-not-found
+  (of-object "GtkRecentChooser")
+  (prop-type "GParamBoolean")
+  (docs "Whether the items pointing to unavailable resources should be displayed")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property show-tips
+  (of-object "GtkRecentChooser")
+  (prop-type "GParamBoolean")
+  (docs "Whether there should be a tooltip on the item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+;; From GtkRecentChooserMenu
+
+(define-property show-icons
   (of-object "GtkRecentChooserMenu")
   (prop-type "GParamBoolean")
-  (docs "Whether to allow multiple items to be selected")
+  (docs "Whether there should be an icon near the item")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property local-only
+(define-property select-multiple
   (of-object "GtkRecentChooserMenu")
   (prop-type "GParamBoolean")
-  (docs "Whether the selected resource(s) should be limited to local file: URIs")
+  (docs "Whether to allow multiple items to be selected")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -28262,19 +29026,37 @@
   (construct-only #f)
 )
 
-(define-property show-not-found
+(define-property limit
   (of-object "GtkRecentChooserMenu")
-  (prop-type "GParamBoolean")
-  (docs "Whether the items pointing to unavailable resources should be displayed")
+  (prop-type "GParamInt")
+  (docs "The maximum number of items to be displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property show-icons
+(define-property recent-manager
   (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamObject")
+  (docs "The RecentManager object to use")
+  (readable #f)
+  (writable #t)
+  (construct-only #t)
+)
+
+(define-property filter
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamObject")
+  (docs "The current filter for selecting which resources are displayed")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property local-only
+  (of-object "GtkRecentChooserMenu")
   (prop-type "GParamBoolean")
-  (docs "Whether there should be an icon near the item")
+  (docs "Whether the selected resource(s) should be limited to local file: URIs")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -28289,6 +29071,24 @@
   (construct-only #f)
 )
 
+(define-property show-not-found
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamBoolean")
+  (docs "Whether the items pointing to unavailable resources should be displayed")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property show-tips
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamBoolean")
+  (docs "Whether there should be a tooltip on the item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property user-data
   (of-object "GtkRecentChooserMenu")
   (prop-type "GParamPointer")
@@ -28487,6 +29287,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkRecentChooserMenu")
   (prop-type "GParamUInt")
@@ -28523,6 +29332,42 @@
   (construct-only #f)
 )
 
+(define-property active
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamInt")
+  (docs "The currently selected menu item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property accel-group
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamObject")
+  (docs "The accel group holding accelerators for the menu")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property accel-path
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamString")
+  (docs "An accel path used to conveniently construct accel paths of child items")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property attach-widget
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamObject")
+  (docs "The widget the menu is attached to")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property tearoff-state
   (of-object "GtkRecentChooserMenu")
   (prop-type "GParamBoolean")
@@ -28541,6 +29386,15 @@
   (construct-only #f)
 )
 
+(define-property monitor
+  (of-object "GtkRecentChooserMenu")
+  (prop-type "GParamInt")
+  (docs "The monitor the menu will be popped up on")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property show-numbers
   (of-object "GtkRecentChooserMenu")
   (prop-type "GParamBoolean")
@@ -28552,33 +29406,42 @@
 
 ;; From GtkRecentChooserWidget
 
-(define-property show-tips
+(define-property show-icons
   (of-object "GtkRecentChooserWidget")
   (prop-type "GParamBoolean")
-  (docs "Whether there should be a tooltip on the item")
+  (docs "Whether there should be an icon near the item")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property limit
+(define-property select-multiple
   (of-object "GtkRecentChooserWidget")
-  (prop-type "GParamInt")
-  (docs "The maximum number of items to be displayed")
+  (prop-type "GParamBoolean")
+  (docs "Whether to allow multiple items to be selected")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property filter
+(define-property sort-type
   (of-object "GtkRecentChooserWidget")
-  (prop-type "GParamObject")
-  (docs "The current filter for selecting which resources are displayed")
+  (prop-type "GParamEnum")
+  (docs "The sorting order of the items displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
+(define-property limit
+  (of-object "GtkRecentChooserWidget")
+  (prop-type "GParamInt")
+  (docs "The maximum number of items to be displayed")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property recent-manager
   (of-object "GtkRecentChooserWidget")
   (prop-type "GParamObject")
@@ -28588,10 +29451,10 @@
   (construct-only #t)
 )
 
-(define-property select-multiple
+(define-property filter
   (of-object "GtkRecentChooserWidget")
-  (prop-type "GParamBoolean")
-  (docs "Whether to allow multiple items to be selected")
+  (prop-type "GParamObject")
+  (docs "The current filter for selecting which resources are displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -28606,10 +29469,10 @@
   (construct-only #f)
 )
 
-(define-property sort-type
+(define-property show-private
   (of-object "GtkRecentChooserWidget")
-  (prop-type "GParamEnum")
-  (docs "The sorting order of the items displayed")
+  (prop-type "GParamBoolean")
+  (docs "Whether the private items should be displayed")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -28624,24 +29487,15 @@
   (construct-only #f)
 )
 
-(define-property show-icons
+(define-property show-tips
   (of-object "GtkRecentChooserWidget")
   (prop-type "GParamBoolean")
-  (docs "Whether there should be an icon near the item")
+  (docs "Whether there should be a tooltip on the item")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property show-private
-  (of-object "GtkRecentChooserWidget")
-  (prop-type "GParamBoolean")
-  (docs "Whether the private items should be displayed")
-  (readable #t)
-  (writable #t)
-  (construct-only #f)
-)
-
 (define-property user-data
   (of-object "GtkRecentChooserWidget")
   (prop-type "GParamPointer")
@@ -28840,6 +29694,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkRecentChooserWidget")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkRecentChooserWidget")
   (prop-type "GParamUInt")
@@ -29131,6 +29994,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkRuler")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property lower
   (of-object "GtkRuler")
   (prop-type "GParamDouble")
@@ -29385,6 +30257,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkScale")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property update-policy
   (of-object "GtkScale")
   (prop-type "GParamEnum")
@@ -29705,6 +30586,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkScaleButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkScaleButton")
   (prop-type "GParamUInt")
@@ -29813,6 +30703,15 @@
   (construct-only #f)
 )
 
+(define-property orientation
+  (of-object "GtkScaleButton")
+  (prop-type "GParamEnum")
+  (docs "The orientation of the scale")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property value
   (of-object "GtkScaleButton")
   (prop-type "GParamDouble")
@@ -30049,6 +30948,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkScrollbar")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property update-policy
   (of-object "GtkScrollbar")
   (prop-type "GParamEnum")
@@ -30340,6 +31248,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkScrolledWindow")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkScrolledWindow")
   (prop-type "GParamUInt")
@@ -30630,6 +31547,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkSeparator")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 ;; From GtkSeparatorToolItem
 
 (define-property user-data
@@ -30830,6 +31756,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkSeparatorToolItem")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkSeparatorToolItem")
   (prop-type "GParamUInt")
@@ -31015,7 +31950,7 @@
 (define-property gtk-icon-sizes
   (of-object "GtkSettings")
   (prop-type "GParamString")
-  (docs "List of icon sizes (gtk-menu=16,16:gtk-button=20,20...)")
+  (docs "List of icon sizes: gtk-menu=16,16:gtk-button=20,20...")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -31291,24 +32226,78 @@
   (construct-only #f)
 )
 
-(define-property gtk-toolbar-style
+(define-property gtk-recent-files-limit
   (of-object "GtkSettings")
-  (prop-type "GParamEnum")
-  (docs "Whether default toolbars have text only, text and icons, icons only, etc.")
+  (prop-type "GParamInt")
+  (docs "Number of recently used files")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property gtk-toolbar-icon-size
+(define-property gtk-im-module
   (of-object "GtkSettings")
-  (prop-type "GParamEnum")
-  (docs "Size of icons in default toolbars")
+  (prop-type "GParamString")
+  (docs "Which IM module should be used by default")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
+(define-property gtk-recent-files-max-age
+  (of-object "GtkSettings")
+  (prop-type "GParamInt")
+  (docs "Maximum age of recently used files, in days")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property gtk-fontconfig-timestamp
+  (of-object "GtkSettings")
+  (prop-type "GParamInt")
+  (docs "Timestamp of current fontconfig configuration")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property gtk-sound-theme-name
+  (of-object "GtkSettings")
+  (prop-type "GParamString")
+  (docs "XDG sound theme name")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property gtk-enable-input-feedback-sounds
+  (of-object "GtkSettings")
+  (prop-type "GParamBoolean")
+  (docs "Whether to play event sounds as feedback to user input")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property gtk-enable-event-sounds
+  (of-object "GtkSettings")
+  (prop-type "GParamBoolean")
+  (docs "Whether to play any event sounds at all")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property gtk-enable-tooltips
+  (of-object "GtkSettings")
+  (prop-type "GParamBoolean")
+  (docs "Whether tooltips should be shown on widgets")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property gtk-button-images
   (of-object "GtkSettings")
   (prop-type "GParamBoolean")
@@ -31318,6 +32307,24 @@
   (construct-only #f)
 )
 
+(define-property gtk-toolbar-style
+  (of-object "GtkSettings")
+  (prop-type "GParamEnum")
+  (docs "Whether default toolbars have text only, text and icons, icons only, etc.")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property gtk-toolbar-icon-size
+  (of-object "GtkSettings")
+  (prop-type "GParamEnum")
+  (docs "Size of icons in default toolbars")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property gtk-label-select-on-focus
   (of-object "GtkSettings")
   (prop-type "GParamBoolean")
@@ -31339,7 +32346,7 @@
 (define-property gtk-entry-password-hint-timeout
   (of-object "GtkSettings")
   (prop-type "GParamUInt")
-  (docs "How long to show the last inputted character in hidden entries")
+  (docs "How long to show the last input character in hidden entries")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -31577,6 +32584,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkSocket")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkSocket")
   (prop-type "GParamUInt")
@@ -31606,6 +32622,12 @@
 
 ;; From GtkSpinButton
 
+(define-signal value-changed
+  (of-object "GtkSpinButton")
+  (return-type "void")
+  (when "last")
+)
+
 (define-signal input
   (of-object "GtkSpinButton")
   (return-type "gint")
@@ -31621,12 +32643,6 @@
   (when "last")
 )
 
-(define-signal value-changed
-  (of-object "GtkSpinButton")
-  (return-type "void")
-  (when "last")
-)
-
 (define-signal wrapped
   (of-object "GtkSpinButton")
   (return-type "void")
@@ -31840,6 +32856,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkSpinButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property cursor-position
   (of-object "GtkSpinButton")
   (prop-type "GParamInt")
@@ -31975,6 +33000,24 @@
   (construct-only #f)
 )
 
+(define-property overwrite-mode
+  (of-object "GtkSpinButton")
+  (prop-type "GParamBoolean")
+  (docs "Whether new text overwrites existing text")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
+(define-property text-length
+  (of-object "GtkSpinButton")
+  (prop-type "GParamUInt")
+  (docs "Length of the text currently in the entry")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property adjustment
   (of-object "GtkSpinButton")
   (prop-type "GParamObject")
@@ -32267,6 +33310,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkStatusbar")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkStatusbar")
   (prop-type "GParamUInt")
@@ -32661,6 +33713,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkTable")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkTable")
   (prop-type "GParamUInt")
@@ -32933,6 +33994,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkTearoffMenuItem")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkTearoffMenuItem")
   (prop-type "GParamUInt")
@@ -32960,6 +34030,15 @@
   (construct-only #f)
 )
 
+(define-property right-justified
+  (of-object "GtkTearoffMenuItem")
+  (prop-type "GParamBoolean")
+  (docs "Sets whether the menu item appears justified at the right side of a menu bar")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property submenu
   (of-object "GtkTearoffMenuItem")
   (prop-type "GParamObject")
@@ -32969,6 +34048,15 @@
   (construct-only #f)
 )
 
+(define-property accel-path
+  (of-object "GtkTearoffMenuItem")
+  (prop-type "GParamString")
+  (docs "Sets the accelerator path of the menu item")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkTipsQuery
 
 (define-signal start-query
@@ -33204,6 +34292,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkTipsQuery")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property xalign
   (of-object "GtkTipsQuery")
   (prop-type "GParamFloat")
@@ -33644,6 +34741,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkToggleButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkToggleButton")
   (prop-type "GParamUInt")
@@ -34017,6 +35123,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkToolbar")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkToolbar")
   (prop-type "GParamUInt")
@@ -34321,6 +35436,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkToolItem")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkToolItem")
   (prop-type "GParamUInt")
@@ -34581,6 +35705,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkToolButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkToolButton")
   (prop-type "GParamUInt")
@@ -35819,6 +36952,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkTextView")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkTextView")
   (prop-type "GParamUInt")
@@ -36330,6 +37472,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkToggleToolButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkToggleToolButton")
   (prop-type "GParamUInt")
@@ -36822,6 +37973,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkTreeView")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkTreeView")
   (prop-type "GParamUInt")
@@ -37011,6 +38171,15 @@
   (construct-only #f)
 )
 
+(define-property tooltip-column
+  (of-object "GtkTreeView")
+  (prop-type "GParamInt")
+  (docs "The column in the model containing the tooltip texts for the rows")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 ;; From GtkTreeViewColumn
 
 (define-signal clicked
@@ -37455,6 +38624,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkViewport")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkViewport")
   (prop-type "GParamUInt")
@@ -37509,7 +38687,6 @@
   (construct-only #f)
 )
 
-
 ;; From GtkVolumeButton
 
 (define-property user-data
@@ -37710,6 +38887,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkVolumeButton")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkVolumeButton")
   (prop-type "GParamUInt")
@@ -37818,6 +39004,15 @@
   (construct-only #f)
 )
 
+(define-property orientation
+  (of-object "GtkVolumeButton")
+  (prop-type "GParamEnum")
+  (docs "The orientation of the scale")
+  (readable #t)
+  (writable #t)
+  (construct-only #f)
+)
+
 (define-property value
   (of-object "GtkVolumeButton")
   (prop-type "GParamDouble")
@@ -38089,7 +39284,7 @@
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
@@ -38391,6 +39586,15 @@
   )
 )
 
+(define-signal damage-event
+  (of-object "GtkWidget")
+  (return-type "gboolean")
+  (when "last")
+  (parameters
+    '("GdkEventExpose*" "p0")
+  )
+)
+
 (define-signal grab-broken-event
   (of-object "GtkWidget")
   (return-type "gboolean")
@@ -38649,6 +39853,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkWidget")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 ;; From GtkWindow
 
 (define-signal set-focus
@@ -38885,6 +40098,15 @@
   (construct-only #f)
 )
 
+(define-property window
+  (of-object "GtkWindow")
+  (prop-type "GParamObject")
+  (docs "The widget's window if it is realized")
+  (readable #t)
+  (writable #f)
+  (construct-only #f)
+)
+
 (define-property border-width
   (of-object "GtkWindow")
   (prop-type "GParamUInt")
@@ -38930,20 +40152,20 @@
   (construct-only #f)
 )
 
-(define-property startup-id
+(define-property role
   (of-object "GtkWindow")
   (prop-type "GParamString")
-  (docs "Unique startup identifier for the window used by startup-notification")
-  (readable #f)
+  (docs "Unique identifier for the window to be used when restoring a session")
+  (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
-(define-property role
+(define-property startup-id
   (of-object "GtkWindow")
   (prop-type "GParamString")
-  (docs "Unique identifier for the window to be used when restoring a session")
-  (readable #t)
+  (docs "Unique startup identifier for the window used by startup-notification")
+  (readable #f)
   (writable #t)
   (construct-only #f)
 )
