diff --git a/gtk/src/gtk_signals.defs b/gtk/src/gtk_signals.defs
index adacfa9..c9433b5 100644
--- a/gtk/src/gtk_signals.defs
+++ b/gtk/src/gtk_signals.defs
@@ -10358,21 +10358,21 @@
   (construct-only #f)
 )
 
 ;; From GtkClipboard
 
 (define-signal owner-change
   (of-object "GtkClipboard")
   (return-type "void")
   (when "first")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventOwnerChange*" "p0")
   )
 )
 
 ;; From GtkColorButton
 
 (define-signal color-set
   (of-object "GtkColorButton")
   (return-type "void")
   (when "first")
 )
@@ -14258,31 +14258,31 @@
     '("const-gchar*" "p0")
   )
 )
 
 (define-signal icon-press
   (of-object "GtkEntry")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkEntryIconPosition" "p0")
-    '("GdkEvent*" "p1")
+    '("const-GdkEventButton*" "p1")
   )
 )
 
 (define-signal icon-release
   (of-object "GtkEntry")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkEntryIconPosition" "p0")
-    '("GdkEvent*" "p1")
+    '("const-GdkEventButton*" "p1")
   )
 )
 
 (define-property editing-canceled
   (of-object "GtkEntry")
   (prop-type "GParamBoolean")
   (docs "Indicates that editing has been canceled")
   (readable #t)
   (writable #t)
   (construct-only #f)
@@ -25195,21 +25195,21 @@
   (of-object "GtkMenuItem")
   (return-type "void")
   (when "first")
 )
 
 (define-signal toggle-size-request
   (of-object "GtkMenuItem")
   (return-type "void")
   (when "first")
   (parameters
-    '("gpointer" "p0")
+    '("int*" "p0")
   )
 )
 
 (define-signal toggle-size-allocate
   (of-object "GtkMenuItem")
   (return-type "void")
   (when "first")
   (parameters
     '("gint" "p0")
   )
@@ -29960,21 +29960,21 @@
 )
 
 (define-signal status-changed
   (of-object "GtkPrintOperation")
   (return-type "void")
   (when "last")
 )
 
 (define-signal create-custom-widget
   (of-object "GtkPrintOperation")
-  (return-type "GObject*")
+  (return-type "GtkWidget*")
   (when "last")
 )
 
 (define-signal update-custom-widget
   (of-object "GtkPrintOperation")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkWidget*" "p0")
     '("GtkPageSetup*" "p1")
@@ -35292,21 +35292,21 @@
   (docs "The size of the recently used resources list")
   (readable #t)
   (writable #f)
   (construct-only #f)
 )
 
 ;; From GtkScale
 
 (define-signal format-value
   (of-object "GtkScale")
-  (return-type "const-gchar*")
+  (return-type "gchar*")
   (when "last")
   (parameters
     '("gdouble" "p0")
   )
 )
 
 (define-property orientation
   (of-object "GtkScale")
   (prop-type "GParamEnum")
   (docs "The orientation of the orientable")
@@ -37957,21 +37957,21 @@
   (prop-type "GParamString")
   (docs "Name of default font to use")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
 (define-property gtk-icon-sizes
   (of-object "GtkSettings")
   (prop-type "GParamString")
-  (docs "List of icon sizes (gtk-menu=16,16:gtk-button=20,20...")
+  (docs "List of icon sizes: gtk-menu=16,16:gtk-button=20,20...")
   (readable #t)
   (writable #t)
   (construct-only #f)
 )
 
 (define-property gtk-modules
   (of-object "GtkSettings")
   (prop-type "GParamString")
   (docs "List of currently active GTK modules")
   (readable #t)
@@ -39173,21 +39173,21 @@
   (of-object "GtkSpinButton")
   (return-type "void")
   (when "last")
 )
 
 (define-signal input
   (of-object "GtkSpinButton")
   (return-type "gint")
   (when "last")
   (parameters
-    '("gpointer" "p0")
+    '("double*" "p0")
   )
 )
 
 (define-signal output
   (of-object "GtkSpinButton")
   (return-type "gboolean")
   (when "last")
 )
 
 (define-signal wrapped
@@ -40370,39 +40370,39 @@
   (parameters
     '("gint" "p0")
   )
 )
 
 (define-signal button-press-event
   (of-object "GtkStatusIcon")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal button-release-event
   (of-object "GtkStatusIcon")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal scroll-event
   (of-object "GtkStatusIcon")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventScroll*" "p0")
   )
 )
 
 (define-signal query-tooltip
   (of-object "GtkStatusIcon")
   (return-type "gboolean")
   (when "last")
   (parameters
     '("gint" "p0")
     '("gint" "p1")
@@ -40410,21 +40410,21 @@
     '("GtkTooltip*" "p3")
   )
 )
 
 (define-signal popup-menu
   (of-object "GtkStatusIcon")
   (return-type "void")
   (when "first")
   (parameters
     '("guint" "p0")
-    '("guint" "p1")
+    '("guint32" "p1")
   )
 )
 
 (define-signal activate
   (of-object "GtkStatusIcon")
   (return-type "void")
   (when "first")
 )
 
 (define-property pixbuf
@@ -44336,53 +44336,53 @@
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
 )
 
 (define-signal mark-set
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
   (parameters
-    '("GtkTextIter*" "p0")
+    '("const-GtkTextIter*" "p0")
     '("GtkTextMark*" "p1")
   )
 )
 
 (define-signal mark-deleted
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkTextMark*" "p0")
   )
 )
 
 (define-signal apply-tag
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal remove-tag
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkTextTag*" "p0")
-    '("GtkTextIter*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p1")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-signal begin-user-action
   (of-object "GtkTextBuffer")
   (return-type "void")
   (when "last")
 )
 
 (define-signal end-user-action
@@ -44476,21 +44476,21 @@
 
 ;; From GtkTextTag
 
 (define-signal event
   (of-object "GtkTextTag")
   (return-type "gboolean")
   (when "last")
   (parameters
     '("GObject*" "p0")
     '("GdkEvent*" "p1")
-    '("GtkTextIter*" "p2")
+    '("const-GtkTextIter*" "p2")
   )
 )
 
 (define-property name
   (of-object "GtkTextTag")
   (prop-type "GParamString")
   (docs "Name used to refer to the text tag. NULL for anonymous tags")
   (readable #t)
   (writable #t)
   (construct-only #t)
@@ -46369,21 +46369,21 @@
   )
 )
 
 (define-signal rows-reordered
   (of-object "GtkTreeModel")
   (return-type "void")
   (when "first")
   (parameters
     '("GtkTreePath*" "p0")
     '("GtkTreeIter*" "p1")
-    '("gpointer" "p2")
+    '("gint*" "p2")
   )
 )
 
 ;; From GtkTreeSelection
 
 (define-signal changed
   (of-object "GtkTreeSelection")
   (return-type "void")
   (when "first")
 )
@@ -48321,30 +48321,30 @@
   (parameters
     '("gboolean" "p0")
   )
 )
 
 (define-signal child-notify
   (of-object "GtkWidget")
   (return-type "void")
   (when "first")
   (parameters
-    '("GParam" "p0")
+    '("GParamSpec*" "p0")
   )
 )
 
 (define-signal draw
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("CairoContext*" "p0")
+    '("cairo_t*" "p0")
   )
 )
 
 (define-signal mnemonic-activate
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
     '("gboolean" "p0")
   )
@@ -48399,183 +48399,183 @@
   (parameters
     '("GdkEvent*" "p0")
   )
 )
 
 (define-signal button-press-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal button-release-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventButton*" "p0")
   )
 )
 
 (define-signal scroll-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventScroll*" "p0")
   )
 )
 
 (define-signal motion-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventMotion*" "p0")
   )
 )
 
 (define-signal delete-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal destroy-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal key-press-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventKey*" "p0")
   )
 )
 
 (define-signal key-release-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventKey*" "p0")
   )
 )
 
 (define-signal enter-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventCrossing*" "p0")
   )
 )
 
 (define-signal leave-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventCrossing*" "p0")
   )
 )
 
 (define-signal configure-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventConfigure*" "p0")
   )
 )
 
 (define-signal focus-in-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventFocus*" "p0")
   )
 )
 
 (define-signal focus-out-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventFocus*" "p0")
   )
 )
 
 (define-signal map-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal unmap-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventAny*" "p0")
   )
 )
 
 (define-signal property-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventProperty*" "p0")
   )
 )
 
 (define-signal selection-clear-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventSelection*" "p0")
   )
 )
 
 (define-signal selection-request-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventSelection*" "p0")
   )
 )
 
 (define-signal selection-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventSelection*" "p0")
   )
 )
 
 (define-signal selection-received
   (of-object "GtkWidget")
   (return-type "void")
   (when "last")
   (parameters
     '("GtkSelectionData*" "p0")
     '("guint" "p1")
@@ -48591,30 +48591,30 @@
     '("guint" "p1")
     '("guint" "p2")
   )
 )
 
 (define-signal proximity-in-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventProximity*" "p0")
   )
 )
 
 (define-signal proximity-out-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventProximity*" "p0")
   )
 )
 
 (define-signal drag-leave
   (of-object "GtkWidget")
   (return-type "void")
   (when "last")
   (parameters
     '("GdkDragContext*" "p0")
     '("guint" "p1")
@@ -48706,57 +48706,57 @@
     '("guint" "p4")
     '("guint" "p5")
   )
 )
 
 (define-signal visibility-notify-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventVisibility*" "p0")
   )
 )
 
 (define-signal client-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventClient*" "p0")
   )
 )
 
 (define-signal window-state-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventWindowState*" "p0")
   )
 )
 
 (define-signal damage-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventExpose*" "p0")
   )
 )
 
 (define-signal grab-broken-event
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("GdkEvent*" "p0")
+    '("GdkEventGrabBroken*" "p0")
   )
 )
 
 (define-signal query-tooltip
   (of-object "GtkWidget")
   (return-type "gboolean")
   (when "last")
   (parameters
     '("gint" "p0")
     '("gint" "p1")
