--- ./../../gdk/src/gdk_methods.defs.orig	2018-06-26 17:10:39.139231000 +0200
+++ ./../../gdk/src/gdk_methods.defs	2018-06-26 17:17:17.130209478 +0200
@@ -2243,14 +2243,17 @@
   (of-object "GdkDragContext")
   (c-name "gdk_drag_context_get_selected_action")
   (return-type "GdkDragAction")
 )
 
-(define-method is_unique
-  (of-object "GdkDragAction")
+; h2def.py thinks that this is a method and that GdkDragAction is an object.
+(define-function gdk_drag_action_is_unique
   (c-name "gdk_drag_action_is_unique")
   (return-type "gboolean")
+  (parameters
+    '("GdkDragAction" "action")
+  )
 )
 
 (define-function gdk_drag_begin
   (c-name "gdk_drag_begin")
   (return-type "GdkDragContext*")
