--- ./../../gdk/src/gdk_signals.defs.orig	2020-08-10 15:54:05.615826709 +0200
+++ ./../../gdk/src/gdk_signals.defs	2020-08-10 15:54:50.182765035 +0200
@@ -1005,11 +1005,11 @@
 (define-signal event
   (of-object "GdkSurface")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("gpointer" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
 (define-signal enter-monitor
   (of-object "GdkSurface")
@@ -1093,11 +1093,11 @@
 (define-signal compute-size
   (of-object "GdkToplevel")
   (return-type "void")
   (flags "Run Last")
   (parameters
-    '("GdkToplevelSize" "p0")
+    '("GdkToplevelSize*" "p0")
   )
 )
 
 (define-property decorated
   (of-object "GdkToplevel")
