--- ./../../gdk/src/gdk_signals.defs.orig	2020-05-21 15:28:16.237314196 +0200
+++ ./../../gdk/src/gdk_signals.defs	2020-05-21 15:38:50.956598646 +0200
@@ -1043,11 +1043,11 @@
 (define-signal event
   (of-object "GdkSurface")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("gpointer" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
 (define-signal enter-monitor
   (of-object "GdkSurface")
