--- tools/gen_scripts/../../gdk/src/gdk_signals.defs.orig	2023-05-05 11:51:35.967589875 +0200
+++ tools/gen_scripts/../../gdk/src/gdk_signals.defs	2023-05-05 12:43:34.512298451 +0200
@@ -533,11 +533,11 @@
 (define-signal compute-size
   (of-object "GdkDragSurface")
   (return-type "void")
   (flags "Run Last")
   (parameters
-    '("GdkDragSurfaceSize" "p0")
+    '("GdkDragSurfaceSize*" "p0")
   )
 )
 
 ;; From GdkDrawContext
 
@@ -1001,11 +1001,11 @@
 (define-signal event
   (of-object "GdkSurface")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("gpointer" "p0")
+    '("GdkEvent*" "p0")
   )
 )
 
 (define-signal layout
   (of-object "GdkSurface")
@@ -1148,11 +1148,11 @@
 (define-signal compute-size
   (of-object "GdkToplevel")
   (return-type "void")
   (flags "Run Last")
   (parameters
-    '("GdkToplevelSize" "p0")
+    '("GdkToplevelSize*" "p0")
   )
 )
 
 (define-property decorated
   (of-object "GdkToplevel")
